<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.1.189">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>BS4017: High Throughput Bioinformatics - 2&nbsp; Genomic Sequencing and Databases</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../chapters/week3.html" rel="next">
<link href="../chapters/week1.html" rel="prev">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../site_libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>
<script async="" src="https://hypothes.is/embed.js"></script>

  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
    <div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Genomic Sequencing and Databases</span></h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../">BS4017: High Throughput Bioinformatics</a> 
        <div class="sidebar-tools-main tools-wide">
    <a href="https://github.com/kevfo/BS4017_Notes" title="Source Code" class="sidebar-tool px-1"><i class="bi bi-github"></i></a>
    <a href="" title="Share" id="sidebar-tool-dropdown-0" class="sidebar-tool dropdown-toggle px-1" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi bi-share"></i></a>
    <ul class="dropdown-menu" aria-labelledby="sidebar-tool-dropdown-0">
        <li>
          <a class="dropdown-item sidebar-tools-main-item" href="https://twitter.com/intent/tweet?url=|url|">
            <i class="bi bi-bi-twitter pe-1"></i>
          Twitter
          </a>
        </li>
        <li>
          <a class="dropdown-item sidebar-tools-main-item" href="https://www.facebook.com/sharer/sharer.php?u=|url|">
            <i class="bi bi-bi-facebook pe-1"></i>
          Facebook
          </a>
        </li>
    </ul>
  <a href="" class="quarto-color-scheme-toggle sidebar-tool" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
  <a href="" class="quarto-reader-toggle sidebar-tool" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
    </div>
      </div>
      <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link">Preface</a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true"><strong>PART I: LECTURE NOTES</strong></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/week1.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Introduction to Linux and the Command Line (untested)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/week2.html" class="sidebar-item-text sidebar-link active"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Genomic Sequencing and Databases</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/week3.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Data Preprocessing and Quality Control</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/week4.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title"><em>De Novo</em> Sequencing</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/week5.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Annotation</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/week6.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Transcriptomes: Experiment Design and RNA Sequence Data Processing</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/week7.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Transcriptomes: Differential Expression to Enrichment Analyses</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/week8.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Transcriptomes: Network Analysis</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#sequencing-fundamentals" id="toc-sequencing-fundamentals" class="nav-link active" data-scroll-target="#sequencing-fundamentals"><span class="toc-section-number">2.1</span>  Sequencing Fundamentals</a></li>
  <li><a href="#illumina-hiseq-and-novaseq" id="toc-illumina-hiseq-and-novaseq" class="nav-link" data-scroll-target="#illumina-hiseq-and-novaseq"><span class="toc-section-number">2.2</span>  Illumina HiSeq and NovaSeq</a>
  <ul class="collapse">
  <li><a href="#typical-illumina-read-structure" id="toc-typical-illumina-read-structure" class="nav-link" data-scroll-target="#typical-illumina-read-structure"><span class="toc-section-number">2.2.1</span>  Typical Illumina Read Structure</a></li>
  <li><a href="#single-end-paired-end-and-mate-pairs" id="toc-single-end-paired-end-and-mate-pairs" class="nav-link" data-scroll-target="#single-end-paired-end-and-mate-pairs"><span class="toc-section-number">2.2.2</span>  Single End, Paired End, and Mate-Pairs</a></li>
  <li><a href="#barcoding" id="toc-barcoding" class="nav-link" data-scroll-target="#barcoding"><span class="toc-section-number">2.2.3</span>  Barcoding</a></li>
  <li><a href="#fastq-formats" id="toc-fastq-formats" class="nav-link" data-scroll-target="#fastq-formats"><span class="toc-section-number">2.2.4</span>  FASTQ Formats</a></li>
  </ul></li>
  <li><a href="#pacbio-sequencing" id="toc-pacbio-sequencing" class="nav-link" data-scroll-target="#pacbio-sequencing"><span class="toc-section-number">2.3</span>  PacBio Sequencing</a>
  <ul class="collapse">
  <li><a href="#pacbio-read-formats" id="toc-pacbio-read-formats" class="nav-link" data-scroll-target="#pacbio-read-formats"><span class="toc-section-number">2.3.1</span>  PacBio Read Formats</a></li>
  <li><a href="#hifi-reads" id="toc-hifi-reads" class="nav-link" data-scroll-target="#hifi-reads"><span class="toc-section-number">2.3.2</span>  HiFi Reads</a></li>
  </ul></li>
  <li><a href="#oxford-nanopore" id="toc-oxford-nanopore" class="nav-link" data-scroll-target="#oxford-nanopore"><span class="toc-section-number">2.4</span>  Oxford Nanopore</a>
  <ul class="collapse">
  <li><a href="#uses-for-long-dna-reads" id="toc-uses-for-long-dna-reads" class="nav-link" data-scroll-target="#uses-for-long-dna-reads"><span class="toc-section-number">2.4.1</span>  Uses for Long DNA Reads</a></li>
  </ul></li>
  <li><a href="#ion-torrent" id="toc-ion-torrent" class="nav-link" data-scroll-target="#ion-torrent"><span class="toc-section-number">2.5</span>  Ion Torrent</a></li>
  <li><a href="#bgiseq-and-mgiseq" id="toc-bgiseq-and-mgiseq" class="nav-link" data-scroll-target="#bgiseq-and-mgiseq"><span class="toc-section-number">2.6</span>  BGISEQ and MGISEQ</a></li>
  <li><a href="#hi-c-chip-seq-10x-and-bisulfite-sequencing" id="toc-hi-c-chip-seq-10x-and-bisulfite-sequencing" class="nav-link" data-scroll-target="#hi-c-chip-seq-10x-and-bisulfite-sequencing"><span class="toc-section-number">2.7</span>  Hi-C, ChiP-seq, 10x, and Bisulfite Sequencing</a>
  <ul class="collapse">
  <li><a href="#hi-c" id="toc-hi-c" class="nav-link" data-scroll-target="#hi-c"><span class="toc-section-number">2.7.1</span>  Hi-C</a></li>
  <li><a href="#x-genomics" id="toc-x-genomics" class="nav-link" data-scroll-target="#x-genomics"><span class="toc-section-number">2.7.2</span>  10x Genomics</a></li>
  <li><a href="#bisulfite-sequencing" id="toc-bisulfite-sequencing" class="nav-link" data-scroll-target="#bisulfite-sequencing"><span class="toc-section-number">2.7.3</span>  Bisulfite Sequencing</a></li>
  <li><a href="#chromatin-immunoprecipitation" id="toc-chromatin-immunoprecipitation" class="nav-link" data-scroll-target="#chromatin-immunoprecipitation"><span class="toc-section-number">2.7.4</span>  Chromatin Immunoprecipitation</a></li>
  </ul></li>
  <li><a href="#online-data-repositories-for-sequenced-data" id="toc-online-data-repositories-for-sequenced-data" class="nav-link" data-scroll-target="#online-data-repositories-for-sequenced-data"><span class="toc-section-number">2.8</span>  Online Data Repositories for Sequenced Data</a>
  <ul class="collapse">
  <li><a href="#sequence-read-archive-i.e.-sra" id="toc-sequence-read-archive-i.e.-sra" class="nav-link" data-scroll-target="#sequence-read-archive-i.e.-sra"><span class="toc-section-number">2.8.1</span>  Sequence Read Archive (i.e., SRA)</a></li>
  <li><a href="#ncbi-genomes" id="toc-ncbi-genomes" class="nav-link" data-scroll-target="#ncbi-genomes"><span class="toc-section-number">2.8.2</span>  NCBI Genomes</a></li>
  <li><a href="#ebi-ensembl-project" id="toc-ebi-ensembl-project" class="nav-link" data-scroll-target="#ebi-ensembl-project"><span class="toc-section-number">2.8.3</span>  EBI: Ensembl Project</a></li>
  </ul></li>
  </ul>
<div class="toc-actions"><div><i class="bi bi-github"></i></div><div class="action-links"><p><a href="https://github.com/kevfo/BS4017_Notes/edit/main/chapters/week2.qmd" class="toc-action">Edit this page</a></p><p><a href="https://github.com/kevfo/BS4017_Notes/issues/new" class="toc-action">Report an issue</a></p></div></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title d-none d-lg-block"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Genomic Sequencing and Databases</span></h1>
</div>



<div class="quarto-title-meta">

    
    
  </div>
  

</header>

<p>Modern day genomic sequencing began with Sanger sequencing.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../imgs/week2/sanger.png" class="img-fluid figure-img" width="304"></p>
<p></p><figcaption class="figure-caption">How Sanger Sequencing Works</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p><strong>Sanger sequencing</strong>, created in 1977, is a method used to read the genetic code of living things. It’s like reading a very long sentence written in a language that cells use. This method can read up to 800 to 1000 ‘letters’ of this genetic sentence in one go. Think of it as reading a paragraph with 800 to 1000 words! And now, modern versions of this method can read 96 paragraphs all at once, making it faster and more efficient.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../imgs/week2/data_amount.png" class="img-fluid figure-img" width="456"></p>
<p></p><figcaption class="figure-caption">Amount of Data Generated with Each Machine</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>Starting from 2004, progress in this field has been really fast. The first machinery that were used before that have become old-fashioned and aren’t used as much anymore.</p>
<section id="sequencing-fundamentals" class="level2" data-number="2.1">
<h2 data-number="2.1" class="anchored" data-anchor-id="sequencing-fundamentals"><span class="header-section-number">2.1</span> Sequencing Fundamentals</h2>
<p>There are a few special words we need to know</p>
<ol type="1">
<li><p><strong>Depth</strong></p>
<p>This is like how many times we read the same sentence in the book to be really sure we got it right.</p></li>
<li><p><strong>Coverage</strong></p>
<p>This is like the average number of times we read different sentences in a part of the book.</p></li>
<li><p><strong>Read length</strong></p>
<p>This is just how much of the book we can read at once. It’s like reading a long or short paragraph.</p></li>
</ol>
<p>When we’re reading this book, we want to make sure we cover every part equally, like giving the same attention to every page. This is called <strong>uniform coverage</strong>.</p>
<p>There are two main ways to read the book:</p>
<ol type="1">
<li>One way reads a lot of sentences quickly, but not all of them in detail.</li>
<li>The other way reads fewer sentences, but really understands them well.</li>
</ol>
<p>So, you can choose to read a lot of the book or read less but really understand it deeply.</p>
</section>
<section id="illumina-hiseq-and-novaseq" class="level2" data-number="2.2">
<h2 data-number="2.2" class="anchored" data-anchor-id="illumina-hiseq-and-novaseq"><span class="header-section-number">2.2</span> Illumina HiSeq and NovaSeq</h2>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../imgs/week2/illumina.png" class="img-fluid figure-img" width="440"></p>
<p></p><figcaption class="figure-caption">Illumina RNA Sequencing Machine Models</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>Creating Illumina libraries is like preparing a special recipe for reading DNA.</p>
<p>First, a long DNA strand and break it into smaller pieces. Then, the ends of these pieces are fixed to make them neat and ready. A special tail is then added to one end, like putting a ribbon on a gift.</p>
<p>After that, tiny adapters, like bookends, are attached to both ends of the DNA pieces. This helps scientists to read them later. Only the right-sized pieces are desired, so any pieces that are too big or too small are removed.</p>
<p>Scientists make many copies of these DNA pieces so we have enough to read. Finally, they use a special machine to read the DNA letters one by one, just like flipping through the pages of a book really fast. This helps scientists learn about the DNA’s secrets and how it works!</p>
<section id="typical-illumina-read-structure" class="level3" data-number="2.2.1">
<h3 data-number="2.2.1" class="anchored" data-anchor-id="typical-illumina-read-structure"><span class="header-section-number">2.2.1</span> Typical Illumina Read Structure</h3>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../imgs/week2/readil.png" class="img-fluid figure-img" width="558"></p>
<p></p><figcaption class="figure-caption">Illumina RNA Read Structures</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>A usual Illumina read has the following structure:</p>
<ol type="1">
<li><p><strong>Illumina Forward</strong></p>
<p>Imagine starting an exciting adventure book. The “Illumina Forward” is like the first page of the book, where the story begins. It’s the beginning point of our DNA reading journey.</p></li>
<li><p><strong>Forward Target Primer</strong></p>
<p>Think of this like a special bookmark in the adventure book. It’s a tiny piece that helps find the exact spot to start reading about in the DNA. It’s like telling the story, “Hey, start reading right here!”</p></li>
<li><p><strong>Target</strong></p>
<p>Picture this as a hidden treasure in the adventure story. The “target” is the part of the DNA that scientists really want to learn about. It’s like the exciting part of the story that holds a secret they’re curious about.</p></li>
<li><p><strong>Reverse Target Primer</strong></p>
<p>Just like the “Forward Target Primer” helps scientists find the beginning, this is like another bookmark that helps scientists find the end of the part we’re interested in. It’s like saying, “Okay, stop reading here!”</p></li>
<li><p><strong>Illumina Reverse</strong></p>
<p>This is like reaching the last page of a book. The Illumina Reverse marks the end of our reading journey for this section of DNA, letting scientists know that we’ve completed the sequence.</p></li>
</ol>
</section>
<section id="single-end-paired-end-and-mate-pairs" class="level3" data-number="2.2.2">
<h3 data-number="2.2.2" class="anchored" data-anchor-id="single-end-paired-end-and-mate-pairs"><span class="header-section-number">2.2.2</span> Single End, Paired End, and Mate-Pairs</h3>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../imgs/week2/matepair.png" class="img-fluid figure-img" width="570"></p>
<p></p><figcaption class="figure-caption">Example of Mate Pair Construction to Construct a Full Genome</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>Illumina is a method for reading DNA that’s like solving different puzzles. Imagine having pieces of a jigsaw puzzle.</p>
<p>First, the scientist can look at just one side of each piece to see what’s there (i.e., <strong>single end</strong>). Or they can look at both sides of the pieces, but not the part in the middle (i.e., <strong>paired end</strong>). There’s also a special kind where they link bigger puzzle pieces together (i.e., <strong>mate-pair</strong>) to understand them better.</p>
<p>In the end, no matter which puzzle they choose, they get to read both sides and find out the average distance between them. These days, the most popular puzzle is the paired end, while the mate-pair puzzle is not used much.</p>
<p>Also, the smaller the puzzle piece, the harder it is to figure out where it fits in the bigger picture of the DNA. But having two sets of puzzle pieces helps scientists figure out where they go in the DNA picture more accurately.</p>
</section>
<section id="barcoding" class="level3" data-number="2.2.3">
<h3 data-number="2.2.3" class="anchored" data-anchor-id="barcoding"><span class="header-section-number">2.2.3</span> Barcoding</h3>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../imgs/week2/barcode.png" class="img-fluid figure-img" width="579"></p>
<p></p><figcaption class="figure-caption">Example of How a Barcode Can be Used</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>Illumina sequencing can use something called **barcodes*&amp;, which are like special tags added to each sequence. Think of them like labels on different books.</p>
<p>Imagine a library of books where and each book has a unique label. In the same way, the sequences get their own labels. If the labels are longer, it’s like having more specific tags for each book. This means you can put even more sequences together for reading, like having a huge shelf of books with similar tags.</p>
<p>For instance, a technology called <em>10x Genomics</em> uses these longer labels to group sequences together, especially when looking at individual cells.</p>
</section>
<section id="fastq-formats" class="level3" data-number="2.2.4">
<h3 data-number="2.2.4" class="anchored" data-anchor-id="fastq-formats"><span class="header-section-number">2.2.4</span> FASTQ Formats</h3>
<p>Illumina returns data in a special text format called <strong>FASTQ</strong>. Imagine it’s like a recipe card for reading DNA. This card has four parts for each piece of DNA it reads:</p>
<ol type="1">
<li><p>The first part is like a name tag. It tells us which piece of DNA we’re looking at. Imagine it’s like a label on a box that says what’s inside.</p></li>
<li><p>The second part is the actual DNA sequence, like a secret code of letters (A, T, C, G, N). It’s like a coded message that we need to decode.</p></li>
<li><p>The third part is a repeat of the name tag. It’s like someone saying, “Hey, this is still the same box we talked about earlier.”</p></li>
<li><p>The fourth part is a set of quality codes. This helps us know how sure we are about each letter in the DNA sequence. It’s like having a confidence score for each letter.</p></li>
</ol>
<p>And these quality codes are kind of like a secret code too, but they’re similar to something called <strong>Phred</strong>. It’s like a scale that tells us how reliable each letter is in the DNA code, just like how we might trust different people’s opinions more or less.</p>
<section id="phred-quality-scores" class="level4" data-number="2.2.4.1">
<h4 data-number="2.2.4.1" class="anchored" data-anchor-id="phred-quality-scores"><span class="header-section-number">2.2.4.1</span> Phred Quality Scores</h4>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../imgs/week2/phred.png" class="img-fluid figure-img" width="417"></p>
<p></p><figcaption class="figure-caption">Phred Score Meanings</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>The quality <span class="math inline">\(Q\)</span> is calculated using the following formula:</p>
<p><span class="math display">\[\begin{equation}
  Q = -10\log_{10}P
\end{equation}\]</span></p>
<p>Though, the scores are reported in something called <strong>ASCII</strong> to save space.</p>
</section>
</section>
</section>
<section id="pacbio-sequencing" class="level2" data-number="2.3">
<h2 data-number="2.3" class="anchored" data-anchor-id="pacbio-sequencing"><span class="header-section-number">2.3</span> PacBio Sequencing</h2>
<p>PacBio sequencing is a way to read DNA that’s like watching a movie frame by frame.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../imgs/week2/pacbio.png" class="img-fluid figure-img" width="425"></p>
<p></p><figcaption class="figure-caption">How PacBio Sequencing Works</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>Imagine a movie reel, but instead of film frames, there are DNA pieces. First, a special machine copies the DNA, making many identical pieces. These pieces are attached to a surface, like putting stickers on a wall.</p>
<p>Then, a tiny camera watches as a machine adds one letter at a time to the DNA chain. It’s like watching someone write a story, but in DNA language. The machine records this process, and scientists can use it to figure out the DNA’s secrets. This method is special because it can read long pieces of DNA in one go, like reading long sentences without stopping.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../imgs/week2/pacbio2.png" class="img-fluid figure-img" width="504"></p>
<p></p><figcaption class="figure-caption">PacBio Sequencing Components</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>PacBio sequencing works a bit like making a detailed copy of a story. The scientists start with a special DNA template called an <strong>SMRTbell</strong>. Imagine this template as the outline of the story. They put it on a surface and use a special machine to make copies of it. These copies are like drafts of the story. Then, a tiny helper called a “polymerase” comes in and reads the story, one letter at a time. It’s like reading the book aloud to remember every detail.</p>
<p>But here’s the cool part: the machine doesn’t just read the story once. It goes over it several times, each time reading a bit more. These shorter readings are called <strong>subreads</strong>. Think of them as reading a book chapter by chapter.</p>
<p>Now, after all these readings, the scientists put everything together like a puzzle. It’s like taking all those drafts and arranging them to get the complete story. This final version is called a <strong>Circular Consensus Sequence</strong>, which is like having the perfect version of the story after making sure all the words are correct. This method helps us read long pieces of DNA with a high level of accuracy, just like getting the full story right.</p>
<section id="pacbio-read-formats" class="level3" data-number="2.3.1">
<h3 data-number="2.3.1" class="anchored" data-anchor-id="pacbio-read-formats"><span class="header-section-number">2.3.1</span> PacBio Read Formats</h3>
<p>When PacBio finishes reading DNA, it’s like recording a video of the process. This video is saved as a “.mov” file, kind of like how you save a video on your phone. But to understand the DNA story better, scientists need to do more things. They use special software called “SMRT Tools,” which PacBio made and shared with everyone. With this software, they can do different tasks, like taking off the starting and ending parts of the video (SMRT bell adapters), and pulling out the most important parts, which are like key scenes in the movie (Circular Consensus Reads / Subreads).</p>
<p>Once they’ve done all this, they can change the video into a different format, like turning a video into pictures or text. They do this to make it easier to work with. They can turn the video into a “.bam” file, which is like a fancy organizer for the pictures, or a “.fastq” file, which is like turning the video into words. All of this helps scientists understand DNA better and find out its secrets. You can learn more about this software and how it works on the PacBio website.</p>
</section>
<section id="hifi-reads" class="level3" data-number="2.3.2">
<h3 data-number="2.3.2" class="anchored" data-anchor-id="hifi-reads"><span class="header-section-number">2.3.2</span> HiFi Reads</h3>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../imgs/week2/hifi.png" class="img-fluid figure-img" width="478"></p>
<p></p><figcaption class="figure-caption">How HiFi Sequencing Works</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>The newest thing from PacBio is called HiFi sequencing. It’s like reading a really long story in a special way. Imagine having a super-long book with chapters that are 20 pages long.</p>
<p>HiFi sequencing reads these chapters many times and figures out the best version. It’s like asking different people to read the same chapter and then picking the one that’s most accurate. This helps us understand long sections of DNA really well.</p>
</section>
</section>
<section id="oxford-nanopore" class="level2" data-number="2.4">
<h2 data-number="2.4" class="anchored" data-anchor-id="oxford-nanopore"><span class="header-section-number">2.4</span> Oxford Nanopore</h2>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../imgs/week2/nanopore.png" class="img-fluid figure-img" width="280"></p>
<p></p><figcaption class="figure-caption">How Oxford Nanopore Sequencing Works</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>Oxford Nanopore technology is a different way to read DNA. Instead of taking pictures, it watches electric signals over time. Think of it like watching a graph that goes up and down. This graph is called a <strong>squiggle</strong>.</p>
<p>They save this squiggle in a special file format called <strong>fast5</strong>. This file not only keeps the squiggle but also the letters that the computer guessed from the squiggle. It’s like writing down both what the graph looks like and what the scientist thinks it means. With new software they’re making, they can even go back to the squiggle and try to guess the letters again to make sure they got them right. It’s like looking at the graph again and trying to understand it even better.</p>
<section id="uses-for-long-dna-reads" class="level3" data-number="2.4.1">
<h3 data-number="2.4.1" class="anchored" data-anchor-id="uses-for-long-dna-reads"><span class="header-section-number">2.4.1</span> Uses for Long DNA Reads</h3>
<p>Using Oxford Nanopore technology for DNA reading has different strengths depending on how many times the DNA is read.</p>
<p>When they read it a few times (i.e., <strong>low coverage</strong>), it’s like filling in the missing pieces of a jigsaw puzzle or adding more information to an already started story. They can use these extra details to make an existing puzzle more complete (gap-filling), make a short story longer and clearer (scaffolding a short read assembly), or combine different pieces from different puzzles into one big picture (hybrid assembly).</p>
<p>But when they read the DNA many, many times (high coverage), it’s like writing a whole new story from scratch. They don’t need any other clues because they have everything they need. This is called “De novo assembly,” where they put together a complete picture of the DNA just by using these long reads. It’s like creating a new jigsaw puzzle using only the pieces from one box.</p>
</section>
</section>
<section id="ion-torrent" class="level2" data-number="2.5">
<h2 data-number="2.5" class="anchored" data-anchor-id="ion-torrent"><span class="header-section-number">2.5</span> Ion Torrent</h2>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../imgs/week2/iontorrent.png" class="img-fluid figure-img" width="333"></p>
<p></p><figcaption class="figure-caption">A Graph Obtained from an Ion Torrent Machine</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>Ion Torrent sequencing is like a special tool that’s great for certain types of projects. When scientists need to read pieces of DNA that are a bit longer (around 400-600 letters), Ion Torrent is like a superhero.</p>
<p>Imagine reading a longer chapter of a book instead of just a few sentences. This is really helpful for projects like studying tiny living things called microbes that live in different places, like our bodies or the environment. These projects, called microbiome studies, benefit a lot from Ion Torrent because it helps scientists understand these tiny creatures in more detail.</p>
</section>
<section id="bgiseq-and-mgiseq" class="level2" data-number="2.6">
<h2 data-number="2.6" class="anchored" data-anchor-id="bgiseq-and-mgiseq"><span class="header-section-number">2.6</span> BGISEQ and MGISEQ</h2>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../imgs/week2/balls.png" class="img-fluid figure-img" width="503"></p>
<p></p><figcaption class="figure-caption">Nanoballs in BGISEQ and MGISEQ</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>BGI, a genomics institute, created special machines for reading DNA. One of them is called BGISEQ-500, kind of like an older model. It reads two lines of DNA that are 100 letters long each, making a total of around 520 million letters. Then there’s a newer model, MGISEQ-2000. It can read two lines that are 200 letters each, for a total of about 1 trillion letters! These machines were made to compete with another popular DNA reader, Illumina. They were designed to be really affordable, making DNA reading cheaper for everyone. It’s like offering a lower-cost way to explore the secrets of genetics.</p>
</section>
<section id="hi-c-chip-seq-10x-and-bisulfite-sequencing" class="level2" data-number="2.7">
<h2 data-number="2.7" class="anchored" data-anchor-id="hi-c-chip-seq-10x-and-bisulfite-sequencing"><span class="header-section-number">2.7</span> Hi-C, ChiP-seq, 10x, and Bisulfite Sequencing</h2>
<p>There are various ways to investigate specific things about DNA, like its unique features. The main difference comes in how the DNA is prepared in the lab, but the actual reading part uses a common Illumina platform. Think of it like different ways to prepare a special dish using the same cooking equipment. No matter which method is used, they all provide a type of output file called “fastq,” which is like a document containing DNA information. However, the way this information is used can vary depending on the specific method employed.</p>
<section id="hi-c" class="level3" data-number="2.7.1">
<h3 data-number="2.7.1" class="anchored" data-anchor-id="hi-c"><span class="header-section-number">2.7.1</span> Hi-C</h3>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../imgs/week2/hic.png" class="img-fluid figure-img" width="443"></p>
<p></p><figcaption class="figure-caption">How Hi-C Sequencing Works</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>Hi-C sequencing is a technique that helps scientists understand how different parts of DNA are arranged in space. It’s like making a map of how different rooms in a house are connected.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../imgs/week2/hicmap.png" class="img-fluid figure-img" width="476"></p>
<p></p><figcaption class="figure-caption">Hi-C Data</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>They capture this information by studying how different parts of the DNA are close to each other. This technique gives them a picture of how far apart or near different parts of the DNA are along the entire chromosome. It’s like learning about the layout of a house by seeing which rooms are close to each other.</p>
</section>
<section id="x-genomics" class="level3" data-number="2.7.2">
<h3 data-number="2.7.2" class="anchored" data-anchor-id="x-genomics"><span class="header-section-number">2.7.2</span> 10x Genomics</h3>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../imgs/week2/10x.png" class="img-fluid figure-img" width="430"></p>
<p></p><figcaption class="figure-caption">How 10x Genomics Works</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>In 10x Genomics technology, each individual piece of DNA is given a special code, like a secret badge. Think of it like giving every player in a game their own unique mark. This mark is created using tiny gel beads in a special mixture. It’s as if each player gets a distinct symbol, helping scientists keep track of different pieces of DNA while they’re doing their research.</p>
<section id="in-unicellular-rna-sequencing" class="level4" data-number="2.7.2.1">
<h4 data-number="2.7.2.1" class="anchored" data-anchor-id="in-unicellular-rna-sequencing"><span class="header-section-number">2.7.2.1</span> In Unicellular RNA Sequencing</h4>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../imgs/week2/10xrna.png" class="img-fluid figure-img" width="381"></p>
<p></p><figcaption class="figure-caption">10x Genomics for RNA Sequencing</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>When using 10x Genomics for single-cell RNA sequencing, there’s a twist in how it works. Instead of using big pieces of DNA, they focus on individual cells. Imagine each cell is like a small character in a story. They put one cell into each tiny gel bead, kind of like each character in their own small bubble. Inside these bubbles, the cell’s RNA, which is like its story, is turned into a special type of DNA called cDNA. This process is like translating the cell’s story into a new language. This way, scientists can study the stories of many individual cells all at once, and see how they’re different or similar.</p>
</section>
</section>
<section id="bisulfite-sequencing" class="level3" data-number="2.7.3">
<h3 data-number="2.7.3" class="anchored" data-anchor-id="bisulfite-sequencing"><span class="header-section-number">2.7.3</span> Bisulfite Sequencing</h3>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../imgs/week2/bisulfite.png" class="img-fluid figure-img" width="428"></p>
<p></p><figcaption class="figure-caption">How Bisulfite Sequencing Works</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>In bisulfite sequencing, DNA is treated with a special chemical called bisulfite. This chemical changes some parts of the DNA. Imagine it’s like using a magic potion on a drawing. When applied to the DNA, bisulfite changes cytosine to uracil, but it doesn’t affect 5-methylcytosine. It’s like turning some parts of the drawing into a new color while leaving other parts the same. This helps scientists understand which parts of the DNA have certain molecules attached to them.</p>
</section>
<section id="chromatin-immunoprecipitation" class="level3" data-number="2.7.4">
<h3 data-number="2.7.4" class="anchored" data-anchor-id="chromatin-immunoprecipitation"><span class="header-section-number">2.7.4</span> Chromatin Immunoprecipitation</h3>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../imgs/week2/chip.png" class="img-fluid figure-img" width="204"></p>
<p></p><figcaption class="figure-caption">How ChIP Works</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p><strong>Ch</strong>romatin <strong>i</strong>mmuno<strong>p</strong>recipitation (i.e., <strong>ChIP</strong>) is a method used to study how proteins interact with DNA. Think of it as a way to find out which proteins are hanging out with specific parts of DNA. Here’s how it works:</p>
<ol type="1">
<li><p>First, scientists fix the proteins they’re interested in, like transcription factors, to the DNA using a special chemical called formaldehyde. Imagine it’s like gluing the proteins to certain parts of the DNA.</p></li>
<li><p>Then they carefully take out the DNA and break it into smaller pieces, like breaking a long necklace into smaller beads.</p></li>
<li><p>Next, they use special antibodies that act like magnets to pull out the proteins they’re studying. It’s like using a magnet to pick up certain toys from a pile.</p></li>
<li><p>After that, they heat everything up to undo the gluing caused by formaldehyde. This step is like melting the glue and separating the proteins from the DNA.</p></li>
</ol>
<p>As a result, they end up with DNA fragments that were connected to the proteins they were interested in. It’s like getting clues about which proteins were spending time with specific parts of the DNA. This helps scientists understand how different proteins control and interact with genes and DNA.</p>
</section>
</section>
<section id="online-data-repositories-for-sequenced-data" class="level2" data-number="2.8">
<h2 data-number="2.8" class="anchored" data-anchor-id="online-data-repositories-for-sequenced-data"><span class="header-section-number">2.8</span> Online Data Repositories for Sequenced Data</h2>
<section id="sequence-read-archive-i.e.-sra" class="level3" data-number="2.8.1">
<h3 data-number="2.8.1" class="anchored" data-anchor-id="sequence-read-archive-i.e.-sra"><span class="header-section-number">2.8.1</span> Sequence Read Archive (i.e., SRA)</h3>
<p>The <strong>S</strong>equence <strong>R</strong>ead <strong>A</strong>rchive (i.e., <strong>SRA</strong>) is like a big library where scientists from around the world store their DNA and RNA sequencing data. It’s kind of like a safe place for important information. This repository is taken care of by a group called NIH, in the USA.</p>
<p>Inside the SRA, you’ll find the raw data from sequencing, which is like the original puzzle pieces of DNA or RNA. Think of it as the untouched information from which scientists make discoveries. Both DNA and RNA data are kept here, like storing books of different kinds in the same library.</p>
<p>Many scientific journals require researchers to share their sequencing data in the SRA. This is important because it allows other scientists to check their work and try things out for themselves. It’s like sharing a recipe so others can cook the same dish. Anyone can download this data for free, helping scientists all over the world learn from each other. However, more detailed information like complete genomes and detailed explanations are stored somewhere else. Sometimes scientists need to process the raw data a bit more to make sense of it, like cooking the raw ingredients into a delicious meal.</p>
</section>
<section id="ncbi-genomes" class="level3" data-number="2.8.2">
<h3 data-number="2.8.2" class="anchored" data-anchor-id="ncbi-genomes"><span class="header-section-number">2.8.2</span> NCBI Genomes</h3>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../imgs/week2/ncbi.png" class="img-fluid figure-img" width="408"></p>
<p></p><figcaption class="figure-caption">NCBI Genomes</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>The NCBI Genomes database is like a huge book that holds a lot of important details about different species. It not only tells you what species a living thing belongs to, but it also lets you know if its DNA has been fully read and studied.</p>
<p>Inside this database, you’ll find information about the species’ classification, kind of like its scientific family tree. It’s like knowing which branch of the animal kingdom it belongs to. Additionally, you can find out if the species’ DNA has been completely read and studied in-depth.</p>
<p>However, this database usually contains only the main or reference version of the species’ DNA, and it comes with annotations that tell you where different genes are located. These annotations are stored in a special kind of file called a “.gff” file, which is like a map showing where different treasures (genes) are hidden in the DNA. So, this database is like a treasure trove of genetic information about different species, helping scientists and researchers understand their DNA better.</p>
</section>
<section id="ebi-ensembl-project" class="level3" data-number="2.8.3">
<h3 data-number="2.8.3" class="anchored" data-anchor-id="ebi-ensembl-project"><span class="header-section-number">2.8.3</span> EBI: Ensembl Project</h3>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../imgs/week2/ebi.png" class="img-fluid figure-img" width="334"></p>
<p></p><figcaption class="figure-caption">Ensembl Project Homepage</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>The EBI’s Ensembl Project is like a special tool that helps scientists explore and understand the DNA of different living things. It’s kind of like a map for navigating the genetic information of various species.</p>
<p>This project offers a genome browser that allows researchers to access annotated genomes of species that belong to specific groups in the animal kingdom. There are different sections, like Ensemble Bacteria, Protists, Fungi, Plants, Metazoa, and Vertebrates. It’s like having different shelves in a library for different types of books.</p>
<p>For each species in Ensembl, you’ll find at least the main version of its DNA and information about its genes. Think of it as knowing the basic story of each species. But sometimes, there’s even more information available, like extra chapters in a book. This project is a helpful tool for scientists to study and learn more about the genetics of various living things.</p>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  let localAlternateSentinel = 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } 
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../chapters/week1.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Introduction to Linux and the Command Line (untested)</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../chapters/week3.html" class="pagination-link">
        <span class="nav-page-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Data Preprocessing and Quality Control</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->



</body></html>