<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.1.189">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>BS4017: High Throughput Bioinformatics - 11&nbsp; Population Analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../chapters/week12.html" rel="next">
<link href="../chapters/week10.html" rel="prev">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../site_libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>
<script async="" src="https://hypothes.is/embed.js"></script>


</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
    <div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title"><span class="chapter-number">11</span>&nbsp; <span class="chapter-title">Population Analysis</span></h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../">BS4017: High Throughput Bioinformatics</a> 
        <div class="sidebar-tools-main tools-wide">
    <a href="https://github.com/kevfo/BS4017_Notes" title="Source Code" class="sidebar-tool px-1"><i class="bi bi-github"></i></a>
    <a href="" title="Share" id="sidebar-tool-dropdown-0" class="sidebar-tool dropdown-toggle px-1" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi bi-share"></i></a>
    <ul class="dropdown-menu" aria-labelledby="sidebar-tool-dropdown-0">
        <li>
          <a class="dropdown-item sidebar-tools-main-item" href="https://twitter.com/intent/tweet?url=|url|">
            <i class="bi bi-bi-twitter pe-1"></i>
          Twitter
          </a>
        </li>
        <li>
          <a class="dropdown-item sidebar-tools-main-item" href="https://www.facebook.com/sharer/sharer.php?u=|url|">
            <i class="bi bi-bi-facebook pe-1"></i>
          Facebook
          </a>
        </li>
    </ul>
  <a href="" class="quarto-color-scheme-toggle sidebar-tool" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
  <a href="" class="quarto-reader-toggle sidebar-tool" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
    </div>
      </div>
      <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link">Preface</a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true"><strong>PART I: LECTURE NOTES</strong></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/week1.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Introduction to Linux and the Command Line (untested)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/week2.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Genomic Sequencing and Databases</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/week3.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Data Preprocessing and Quality Control</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/week4.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title"><em>De Novo</em> Sequencing</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/week5.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Annotation</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/week6.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Transcriptomes: Experiment Design and RNA Sequence Data Processing</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/week7.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Transcriptomes: Differential Expression to Enrichment Analyses</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/week8.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Transcriptomes: Network Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/week9.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">SNPs, Structural Variations, and Pan-Genomes</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/week10.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">Re-Sequencing and SNP Calling</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/week11.html" class="sidebar-item-text sidebar-link active"><span class="chapter-number">11</span>&nbsp; <span class="chapter-title">Population Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/week12.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">12</span>&nbsp; <span class="chapter-title">Comparative Genomics</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#key-concepts-and-terms" id="toc-key-concepts-and-terms" class="nav-link active" data-scroll-target="#key-concepts-and-terms"><span class="toc-section-number">11.1</span>  Key Concepts and Terms</a>
  <ul class="collapse">
  <li><a href="#admixture" id="toc-admixture" class="nav-link" data-scroll-target="#admixture"><span class="toc-section-number">11.1.1</span>  Admixture</a></li>
  <li><a href="#deep-coalesence" id="toc-deep-coalesence" class="nav-link" data-scroll-target="#deep-coalesence"><span class="toc-section-number">11.1.2</span>  Deep Coalesence</a></li>
  </ul></li>
  <li><a href="#genetic-distances" id="toc-genetic-distances" class="nav-link" data-scroll-target="#genetic-distances"><span class="toc-section-number">11.2</span>  Genetic Distances</a>
  <ul class="collapse">
  <li><a href="#calculating-genetic-distances" id="toc-calculating-genetic-distances" class="nav-link" data-scroll-target="#calculating-genetic-distances"><span class="toc-section-number">11.2.1</span>  Calculating Genetic Distances</a></li>
  <li><a href="#ibs-distances" id="toc-ibs-distances" class="nav-link" data-scroll-target="#ibs-distances"><span class="toc-section-number">11.2.2</span>  IBS Distances</a></li>
  <li><a href="#tracking-genealogy" id="toc-tracking-genealogy" class="nav-link" data-scroll-target="#tracking-genealogy"><span class="toc-section-number">11.2.3</span>  Tracking Genealogy</a></li>
  </ul></li>
  <li><a href="#species-versus-populations" id="toc-species-versus-populations" class="nav-link" data-scroll-target="#species-versus-populations"><span class="toc-section-number">11.3</span>  Species versus Populations</a>
  <ul class="collapse">
  <li><a href="#tests-for-introgression" id="toc-tests-for-introgression" class="nav-link" data-scroll-target="#tests-for-introgression"><span class="toc-section-number">11.3.1</span>  Tests for Introgression?</a></li>
  <li><a href="#tests-for-directionality" id="toc-tests-for-directionality" class="nav-link" data-scroll-target="#tests-for-directionality"><span class="toc-section-number">11.3.2</span>  Tests for Directionality</a></li>
  </ul></li>
  <li><a href="#population-history" id="toc-population-history" class="nav-link" data-scroll-target="#population-history"><span class="toc-section-number">11.4</span>  Population History</a>
  <ul class="collapse">
  <li><a href="#site-frequency-spectrum" id="toc-site-frequency-spectrum" class="nav-link" data-scroll-target="#site-frequency-spectrum"><span class="toc-section-number">11.4.1</span>  Site Frequency Spectrum</a></li>
  </ul></li>
  <li><a href="#causative-mutations" id="toc-causative-mutations" class="nav-link" data-scroll-target="#causative-mutations"><span class="toc-section-number">11.5</span>  Causative Mutations</a>
  <ul class="collapse">
  <li><a href="#beneficial-mutations" id="toc-beneficial-mutations" class="nav-link" data-scroll-target="#beneficial-mutations"><span class="toc-section-number">11.5.1</span>  Beneficial Mutations</a></li>
  <li><a href="#sweep-analysis" id="toc-sweep-analysis" class="nav-link" data-scroll-target="#sweep-analysis"><span class="toc-section-number">11.5.2</span>  Sweep Analysis</a></li>
  </ul></li>
  </ul>
<div class="toc-actions"><div><i class="bi bi-github"></i></div><div class="action-links"><p><a href="https://github.com/kevfo/BS4017_Notes/edit/main/chapters/week11.qmd" class="toc-action">Edit this page</a></p><p><a href="https://github.com/kevfo/BS4017_Notes/issues/new" class="toc-action">Report an issue</a></p></div></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title d-none d-lg-block"><span class="chapter-number">11</span>&nbsp; <span class="chapter-title">Population Analysis</span></h1>
</div>



<div class="quarto-title-meta">

    
    
  </div>
  

</header>

<p><strong>SNPs</strong>, or <strong>Single Nucleotide Polymorphisms</strong>, are changes that occur in our genetic code over long periods. Imagine them as tiny mutations that happen naturally in our DNA as time goes on. In humans, the mutation rate - the rate at which these changes happen - is around 0.5 to 1 mutation per 100 million base pairs in just one generation. This means about 70 to 80 mutations can occur in each person.</p>
<p>These changes aren’t predictable and happen randomly, and they are quite uncommon. However, when we find the same SNPs shared among different individuals, it gives us clues about their shared ancestors.</p>
<p>Scientists use a theory called <strong>coalescent theory</strong> to understand this. This theory says that any two sets of genes or genomes will always have an ancestor they both come from.</p>
<p>When two people share a lot of SNPs, it means they probably have similar ancestors in their family trees. On the other hand, if their SNPs are different, it shows that their genetic sequences have traveled a longer evolutionary distance apart.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../imgs/week11/mrca.png" class="img-fluid figure-img" width="308"></p>
<p></p><figcaption class="figure-caption">Illustration of a MRCA</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>Scientists can use this information to trace back and find the <strong>most recent common ancestor (MRCA)</strong> shared by two individuals or groups.</p>
<p>The <strong>neutral theory of evolution</strong> suggests that a lot of the changes or mutations happening in our genetic code don’t really affect how an organism survives or thrives. These mutations are considered neutral because they don’t give any advantage or disadvantage in the environment. Imagine these changes as happening randomly, without impacting an individual’s chances of survival.</p>
<p><strong>Drift</strong>, another concept in genetics, is responsible for explaining most of the patterns we see in SNPs (those small genetic differences). Drift refers to random changes in the frequency of certain genetic traits within a population over time.</p>
<p>Most of these SNP patterns we observe in populations are because of shared ancestry. This means that when we look at the differences and similarities in these small genetic differences among people, we can understand how different groups or populations are related to each other.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../imgs/week11/pca_plot.png" class="img-fluid figure-img" width="348"></p>
<p></p><figcaption class="figure-caption">PCA Plot on Human Data</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>To help visualize how populations are structured based on these SNPs, scientists use a method called <strong>Principal Component Analysis (PCA)</strong>. They organize the SNP data into a matrix where each SNP is represented by numbers: 0, 1, or 2 (showing the number of alternative forms of a particular gene). By analyzing this data, they can see that individuals who have similar patterns of these genetic differences are grouped closer together.</p>
<section id="key-concepts-and-terms" class="level2" data-number="11.1">
<h2 data-number="11.1" class="anchored" data-anchor-id="key-concepts-and-terms"><span class="header-section-number">11.1</span> Key Concepts and Terms</h2>
<section id="admixture" class="level3" data-number="11.1.1">
<h3 data-number="11.1.1" class="anchored" data-anchor-id="admixture"><span class="header-section-number">11.1.1</span> Admixture</h3>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../imgs/week11/admixture.png" class="img-fluid figure-img" width="348"></p>
<p></p><figcaption class="figure-caption">Illustration of an Admixture</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p><strong>Admixture</strong> is a concept that helps scientists divide or separate genetic data into groups that represent different ancestral populations. These groups have variations in the frequencies of certain forms of genes (alleles) across their genomes.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../imgs/week11/admixture_task.png" class="img-fluid figure-img" width="472"></p>
<p></p><figcaption class="figure-caption">More on Admixtures</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>In this process, scientists aim to categorize the data into k groups, each representing a population with its unique allele frequencies. This method helps to understand how various genetic variations are distributed among different populations.</p>
<p>Admixture can be thought of as a specific type of Principal Component Analysis (PCA), a statistical technique used to simplify complex data.</p>
</section>
<section id="deep-coalesence" class="level3" data-number="11.1.2">
<h3 data-number="11.1.2" class="anchored" data-anchor-id="deep-coalesence"><span class="header-section-number">11.1.2</span> Deep Coalesence</h3>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../imgs/week11/deep.png" class="img-fluid figure-img" width="186"></p>
<p></p><figcaption class="figure-caption">Grpahic Illustrating Deep Coalesence</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p><strong>Deep coalescence</strong> refers to a situation where a specific genetic variation, like a SNP (Single Nucleotide Polymorphism), existed in a common ancestor of two different species. This means that this genetic difference, represented by the SNP, was present before these species branched off and became distinct from each other.</p>
<p>When scientists identify a SNP that is shared between two different species, it suggests that this particular genetic variation dates back to a time before these species evolved separately. The “deep” aspect refers to the fact that this genetic variation’s origin predates the actual separation of these species into distinct evolutionary paths.</p>
<p><strong>Incomplete lineage</strong> sorting occurs when genetic variations, including SNPs, are inherited from a common ancestor but don’t align neatly with the species’ evolutionary tree. This happens because, over generations, genetic diversity is maintained within a population due to both ancestral genetic differences and random genetic drift. When these variations persist and get passed on to subsequent generations, they might not conform to the species tree we construct based on other evidence, like fossils or morphological features.</p>
<p>Both these phenomena—ancestral polymorphism (genetic differences in a common ancestor) and genetic drift (random changes in genetic frequency)—can cause incomplete lineage sorting. This results in the SNP pattern not following the expected species evolutionary tree because some genetic variations persist across lineages despite the species branching out from a common ancestor.</p>
</section>
</section>
<section id="genetic-distances" class="level2" data-number="11.2">
<h2 data-number="11.2" class="anchored" data-anchor-id="genetic-distances"><span class="header-section-number">11.2</span> Genetic Distances</h2>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../imgs/week11/recombination.png" class="img-fluid figure-img" width="440"></p>
<p></p><figcaption class="figure-caption">Illustration of Recombination</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p><strong>Recombination</strong> is a genetic process where sections of DNA, including SNPs (Single Nucleotide Polymorphisms), are shuffled or exchanged between chromosomes during cell division. This process results in new combinations of genetic variations.</p>
<p>One important aspect related to SNPs and recombination is linkage disequilibrium. <strong>Linkage disequilibrium</strong> refers to the non-random association or correlation of specific alleles at two different locations in the genome within a general population. This means that certain alleles at different locations tend to occur together more frequently than would be expected by chance.</p>
<p>SNPs that are physically close to each other on the genome tend to exhibit higher linkage disequilibrium, meaning they are more likely to be inherited together. However, as the distance between SNPs increases along the genome, the degree of correlation or linkage disequilibrium between them decreases.</p>
<p>The concept of a haplotype block is related to linkage disequilibrium and recombination. A <strong>haplotype block</strong> refers to a region in the genome where specific sets of alleles are inherited together as a result of limited or no history of recombination. In these blocks, certain combinations of genetic variations tend to stay together across generations because they are less likely to be broken apart by the recombination process.</p>
<section id="calculating-genetic-distances" class="level3" data-number="11.2.1">
<h3 data-number="11.2.1" class="anchored" data-anchor-id="calculating-genetic-distances"><span class="header-section-number">11.2.1</span> Calculating Genetic Distances</h3>
<p><strong>Genetic distance</strong> refers to a measure used to understand how closely related two individuals are based on their genetic information, particularly Single Nucleotide Polymorphisms (SNPs).</p>
<p>One way to calculate this relatedness is through the concept of <strong>Identity by State (IBS) distance</strong>. IBS distance looks at long stretches of the genome where two individuals share at least one allele (or genetic variant) at each location. By identifying these continuous regions where the genetic information matches between two individuals, scientists can gauge their genetic relatedness.</p>
<p>Ideally, the best method to determine genetic distance would involve identifying haplotype blocks, which are regions of the genome where specific combinations of alleles tend to be inherited together. Comparing these blocks between two individuals’ genomes would give a precise measure of relatedness. However, this method requires information about the complete sequences of these blocks, which is often not available from standard sequencing data. This limitation arises because the individual reads obtained from sequencing are usually too short to cover entire haplotypes.</p>
<p>As a result, IBS distance calculates relatedness by considering scenarios where at least one of the underlying haplotypes could be shared between two individuals. The longer the matching regions of genetic information (indicating the potential sharing of a common haplotype block), the closer the genetic relatedness between the individuals.</p>
</section>
<section id="ibs-distances" class="level3" data-number="11.2.2">
<h3 data-number="11.2.2" class="anchored" data-anchor-id="ibs-distances"><span class="header-section-number">11.2.2</span> IBS Distances</h3>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../imgs/week11/IBS.png" class="img-fluid figure-img" width="460"></p>
<p></p><figcaption class="figure-caption">Illustration of how IBS Distances Work</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p><strong>Identity by State (IBS) distance</strong> is a method used to measure genetic relatedness between individuals by examining long shared segments of their DNA known as haplotype blocks. These haplotype blocks are regions in the genome where certain combinations of genetic variants tend to be inherited together.</p>
<p>One key characteristic of IBS distance is that as the genetic distance between individuals increases, the length of these shared haplotype blocks tends to decrease. In other words, when two individuals are more distantly related, they are less likely to share long stretches of identical genetic information.</p>
<p>While IBS distance provides a useful measure of genetic relatedness based on these shared haplotype blocks, for a more accurate and detailed understanding of exact relationships between individuals, a closer analysis of these haplotype blocks becomes necessary.</p>
</section>
<section id="tracking-genealogy" class="level3" data-number="11.2.3">
<h3 data-number="11.2.3" class="anchored" data-anchor-id="tracking-genealogy"><span class="header-section-number">11.2.3</span> Tracking Genealogy</h3>
<p>Public companies specializing in genetic testing, like 23andMe, circleDNA, and Nebula Genomics, use various technologies to analyze Single Nucleotide Polymorphisms (SNPs) in individuals’ DNA.</p>
<p>These companies employ different methods such as SNP chips (used by 23andMe), whole exome sequencing (used by circleDNA), and whole-genome sequencing (used by Nebula Genomics) to decode and interpret genetic information. They have amassed a substantial database, collectively having over 25 million people enrolled.</p>
<p>When individuals send in their DNA samples, these companies compare the genetic information in these samples to others in their databases. By doing so, they can identify long segments of shared genetic information known as Identity by State (IBS) stretches, helping to trace relatedness and ancestry among different individuals.</p>
<p>This analysis enables these companies to determine potential relatives by identifying shared genetic markers. Moreover, they can provide insights into an individual’s ethnic background based on the genetic information obtained.</p>
<p>Additionally, these genetic testing services compare an individual’s DNA to known genetic markers associated with certain diseases or specific traits. This helps users understand their potential genetic predispositions to certain health conditions or their likelihood of expressing particular characteristics.</p>
<p>However, it’s essential to note that while these associations between genetic markers and traits exist, they often have relatively weak correlations. This is sometimes referred to as “missing heritability,” where the identified genetic markers do not fully explain the inheritance or development of certain traits or diseases. There are likely other genetic or environmental factors that play significant roles, beyond what is currently identified through these associations</p>
<section id="security-issues" class="level4" data-number="11.2.3.1">
<h4 data-number="11.2.3.1" class="anchored" data-anchor-id="security-issues"><span class="header-section-number">11.2.3.1</span> Security Issues?</h4>
<p>The convenience of genetic testing and tracking genealogy comes with potential security and privacy concerns:</p>
<p>Firstly, when you submit your DNA to a company, your genetic information becomes part of their database. This information, including your SNPs (Single Nucleotide Polymorphisms), can be used to find new connections between your genetic makeup and certain traits or conditions. Big medical companies find large SNP databases incredibly valuable for their research purposes.</p>
<p>There’s a particular interest in identifying “natural gene knock-outs,” which are genetic variations that eliminate the function of a certain gene. Understanding these can shed light on how genes influence traits or diseases.</p>
<p>Moreover, several companies like GEDmatch, MyHeritage, FamilyTreeDNA, and LivingDNA allow individuals to deposit their genetic data and compare it against their databases. This exchange of information can reveal extensive details about an individual’s genetic makeup.</p>
<p>Unfortunately, these databases are susceptible to potential breaches in genetic privacy. Tailor-made data queries can exploit these databases, revealing more information than intended or agreed upon.</p>
<p>As legislation struggles to keep pace with this rapidly evolving field, there’s a concerning gap in regulation to protect individuals’ genetic privacy. The novelty of genetic testing and genealogical tracking means that the legal framework is not yet comprehensive enough to address the risks adequately.</p>
</section>
</section>
</section>
<section id="species-versus-populations" class="level2" data-number="11.3">
<h2 data-number="11.3" class="anchored" data-anchor-id="species-versus-populations"><span class="header-section-number">11.3</span> Species versus Populations</h2>
<p><strong>Species</strong> are distinct groups of organisms that are capable of interbreeding and producing fertile offspring. They are characterized by significant differences in their genetic makeup, physical characteristics, and behaviors. The variation observed between different species tends to be greater than the variation seen within populations of the same species.</p>
<p>The divergence between species arises from evolutionary processes over extended periods. A common ancestor shared by two different species existed far back in time, often millions of years ago. This prolonged separation allowed for substantial genetic and phenotypic differences to accumulate, leading to the development of reproductive barriers.</p>
<p><strong>Reproductive barriers</strong> are mechanisms or traits that prevent two different species from interbreeding successfully or producing fertile offspring. Over time, these barriers become established and can include differences in mating behaviors, genetic incompatibility, physical barriers, or ecological factors.</p>
<p>One way of defining a species is by assessing the presence of reproductive barriers. When enough time has passed for these barriers to evolve and prevent gene flow between populations, scientists consider these populations as distinct species. In simpler terms, if groups of organisms have been separated long enough to develop mechanisms that prevent them from successfully interbreeding, they are typically classified as different species.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../imgs/week11/introgression.png" class="img-fluid figure-img" width="212"></p>
<p></p><figcaption class="figure-caption">Introgression Illustration</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>When individuals from two different species mate before the complete formation of reproductive barriers, a process called “<strong>introgression</strong>” can occur.</p>
<p>Introgression involves the transfer of genetic material between two distinct species through hybridization, where individuals from different species interbreed and produce offspring. In these cases, the genetic material from one species crosses into the gene pool of the other species.</p>
<p>This exchange of genetic material can lead to the incorporation of genes from one species into the genome of the other. Sometimes, this can result in hybrids that possess a mixture of characteristics from both parent species.</p>
<p>The extent and impact of introgression can vary. In some cases, if the hybrid offspring are not reproductively isolated from either parent species, they might continue to breed with one or both parent species. Over time, this process can lead to the spread of genetic material across the species boundary.</p>
<p>Introgression can have significant effects on the genetic diversity and evolutionary trajectories of species. It can potentially influence the adaptation and evolution of species by introducing new genetic variations into their gene pools. Additionally, introgression can blur the lines between distinct species, making it more challenging to delineate boundaries between them based solely on genetic markers</p>
<section id="tests-for-introgression" class="level3" data-number="11.3.1">
<h3 data-number="11.3.1" class="anchored" data-anchor-id="tests-for-introgression"><span class="header-section-number">11.3.1</span> Tests for Introgression?</h3>
<p>Do look at the slides to know more about this (i.e., prof. Jarkko talks about how the F3 introgression test works in more detail on that slide)!</p>
<section id="f3-introgression-test" class="level4" data-number="11.3.1.1">
<h4 data-number="11.3.1.1" class="anchored" data-anchor-id="f3-introgression-test"><span class="header-section-number">11.3.1.1</span> F3 Introgression Test</h4>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../imgs/week11/f3.png" class="img-fluid figure-img" width="482"></p>
<p></p><figcaption class="figure-caption">F3 Introgression Test</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>Positive evidence of introgression can be identified through statistical analyses that assess the patterns of genetic variation within genomes.</p>
<p>When there’s no introgression between two populations or species, the genetic variations such as positive and negative values occur randomly throughout the genome. However, when introgression has taken place, certain statistical methods can reveal a skewed pattern in these genetic variations.</p>
<p>One such method involves calculating a statistic known as the <strong>F3 statistic</strong> and comparing it to an expected score based on the null hypothesis, which assumes random genetic variation. The distribution of this statistic under the null hypothesis can be determined by simulations or random sequences.</p>
<p>In cases where introgression has occurred, there tends to be an excess of negative values compared to positive values when calculating the F3 statistic across the genome. This skewed distribution with a prevalence of highly negative values indicates a signal of introgression.</p>
<p>To further evaluate the significance of this signal, statistical techniques can be applied to assess the likelihood of observing such a pattern under the null hypothesis. This evaluation can yield a p-value, indicating the probability of obtaining the observed pattern if there were no introgression. A low p-value suggests strong evidence against the null hypothesis and supports the presence of introgression.</p>
<p>To determine the direction of gene flow or introgression between two test populations, an additional species, typically an outgroup species (O), is introduced to serve as a reference point for ancestral allele states. This outgroup species is evolutionarily distant from the populations being studied and provides a baseline for understanding genetic changes that occurred before divergence.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../imgs/week11/outgroup.png" class="img-fluid figure-img" width="152"></p>
<p></p><figcaption class="figure-caption">Setup for an Introgression Test</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>The setup involves several components:</p>
<ol type="1">
<li><p><strong>Outgroup species (O)</strong>: This species, like chimpanzees for humans, serves as a reference to identify ancestral allele states. By comparing the genetic sequences of the outgroup species with those of the test populations, scientists can infer the ancestral alleles shared between the outgroup and the test populations before any introgression events.</p></li>
<li><p><strong>Second species (W)</strong>: This species is considered a possible source of introgression. By comparing the genetic sequences of the second species with the test populations, scientists can identify genetic variations that may have been introduced via introgression.</p></li>
<li><p><strong>Test populations (A and B)</strong>: These are the populations being studied for introgression. By comparing their genetic sequences with the outgroup species and the second species, scientists can assess which genetic variations are shared with the outgroup (ancestral) and which are specific to the second species (potential introgression).</p></li>
</ol>
<p>By analyzing the genetic variations shared between the outgroup species, the second species, and the test populations, researchers can deduce the direction of gene flow. If specific genetic variations are present in the second species (W) and also found in test population A but not in test population B, it suggests that gene flow occurred from species W into population A. Conversely, if these variations are found in population B but not in A, it indicates gene flow in the opposite direction.</p>
</section>
</section>
<section id="tests-for-directionality" class="level3" data-number="11.3.2">
<h3 data-number="11.3.2" class="anchored" data-anchor-id="tests-for-directionality"><span class="header-section-number">11.3.2</span> Tests for Directionality</h3>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../imgs/week11/directionality.png" class="img-fluid figure-img" width="358"></p>
<p></p><figcaption class="figure-caption">Testing for Directionality</figcaption><p></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../imgs/week11/directionality2.png" class="img-fluid figure-img" width="369"></p>
<p></p><figcaption class="figure-caption">Testing for Directionality</figcaption><p></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../imgs/week11/directionality3.png" class="img-fluid figure-img" width="509"></p>
<p></p><figcaption class="figure-caption">Testing for Directionality</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>To test for the directionality of introgression or gene flow between populations, scientists often search for Single Nucleotide Polymorphisms (SNPs) that exhibit two derived alleles.</p>
<p><strong>Derived alleles</strong> refer to genetic variants that are not present in the ancestral population or outgroup. When examining SNPs in test populations A and B and comparing them with an outgroup species, if a particular SNP shows two derived alleles (meaning both populations have a unique allele not found in the outgroup), it suggests a potential introgression event.</p>
<p>The key aspect scientists focus on is asymmetry in the distribution of these derived alleles between the two test populations. If, for example, the SNP has derived alleles that are present predominantly or exclusively in population A and not in population B, it indicates a higher likelihood of gene flow or introgression from another source into population A.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../imgs/week11/testing.png" class="img-fluid figure-img" width="376"></p>
<p></p><figcaption class="figure-caption">Testing for Directionality via the ABBA-BABA Test</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>The <strong>ABBA-BABA</strong> test, also known as the <strong>D-statistic</strong> or <strong>Patterson’s D</strong>, is a method used in population genetics to detect introgression or gene flow between populations.</p>
<p>The test involves counting the occurrences of different allele configurations across multiple loci or genomic regions within four different populations or groups (A, B, W, and O). Specifically, researchers count how many times the pattern ABBA occurs versus the pattern BABA across these populations at various genetic loci.</p>
<p>In scenarios where genetic variations are randomly distributed or shaped only by drift (i.e., no gene flow between populations), the difference between the occurrences of ABBA and BABA should be close to zero when calculated across the entire genome.</p>
<p>However, when there’s introgression or shared ancestry between populations, the ABBA and BABA counts will differ significantly from zero. A non-zero or highly non-zero value indicates a deviation from the expected pattern, suggesting potential gene flow or introgression between the studied populations.</p>
</section>
</section>
<section id="population-history" class="level2" data-number="11.4">
<h2 data-number="11.4" class="anchored" data-anchor-id="population-history"><span class="header-section-number">11.4</span> Population History</h2>
<p>The <strong>coalescent theory</strong> is a mathematical model used in population genetics to understand how gene variants sampled from a population trace back to a common ancestor. This theory helps us examine the historical patterns of genetic variation within populations.</p>
<p>Imagine looking backward in time, tracing the ancestry of gene variants present in a population. The coalescent theory explores this genetic history by simulating the merging of alleles, or gene variants, into a single common ancestral copy through a series of random coalescence events.</p>
<p>These coalescence events represent moments when gene variants from different individuals in a population merge into a single common ancestor. The model calculates the time or number of generations it takes for these variants to converge and share a common ancestral copy.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../imgs/week11/path.png" class="img-fluid figure-img" width="302"></p>
<p></p><figcaption class="figure-caption">Illustration of Coalesence Theory</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>One method involves segmenting the individual’s genome into regions with varying levels of <strong>heterozygosity</strong>, which refers to having different versions of a gene in an individual’s genome. Higher heterozygosity indicates regions where the genetic material originated from a more distant common ancestor, representing deeper coalescence events. In contrast, lower heterozygosity regions suggest genetic material that traces back to a more recent common ancestor.</p>
<p>These different segments or regions of varying heterozygosity provide insights into the historical recombination events that occurred in the population’s past. Recombination refers to the shuffling of genetic material during reproduction, resulting in new combinations of genetic variants.</p>
<p>Regions with many splits or changes in heterozygosity levels signify past ancestral recombination events. The number of these recombination events is often linked to the historical population size. Larger populations tend to have more recombination events due to increased genetic diversity and more opportunities for genetic material to mix during reproduction.</p>
<section id="site-frequency-spectrum" class="level3" data-number="11.4.1">
<h3 data-number="11.4.1" class="anchored" data-anchor-id="site-frequency-spectrum"><span class="header-section-number">11.4.1</span> Site Frequency Spectrum</h3>
<p>The <strong>site frequency spectrum (SFS)</strong> is an alternative method in population genetics used to analyze genetic variation within a population. It involves creating a histogram that shows the frequency of a particular allele (often a derived allele) across the population.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../imgs/week11/sfs.png" class="img-fluid figure-img" width="462"></p>
<p></p><figcaption class="figure-caption">Site Frequency Spectrum</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>Ideally, the SFS is created by comparing the frequencies of alleles in the studied population with the ancestral genome. However, if the ancestral genome is not available, the “folded” spectrum is used. The folded spectrum combines the frequencies of derived alleles with their complementary ancestral alleles, which means fixated alleles (those that have completely replaced the ancestral form) and rare SNPs can’t be distinguished. This leads to a reduction in resolution as information about these alleles becomes limited.</p>
<p>Different population histories result in distinct site frequency spectra. The goal with this method is to identify the historical scenario that is most likely to have produced the observed spectrum.</p>
</section>
</section>
<section id="causative-mutations" class="level2" data-number="11.5">
<h2 data-number="11.5" class="anchored" data-anchor-id="causative-mutations"><span class="header-section-number">11.5</span> Causative Mutations</h2>
<p><strong>Causative mutations</strong> refer to genetic alterations that directly impact the function or structure of proteins encoded by specific genes. These mutations can lead to changes in the amino acid sequence of a protein, which might affect its normal function or activity.</p>
<p>When a mutation alters the sequence of amino acids in a protein, it can result in various outcomes. For instance, it might modify the protein’s shape or ability to interact with other molecules, affecting its function. In the most extreme cases, a mutation can introduce a premature STOP codon in the genetic code, causing the protein to be shorter than normal or non-functional.</p>
<p>These mutations that cause changes in protein structure or function are often referred to as “<strong>natural mutants</strong>.” However, it’s important to note that most of these mutations tend to reduce the fitness or health of an organism rather than improve it. In other words, they usually have negative effects on the organism’s ability to survive, reproduce, or function optimally within its environment.</p>
<p>While some mutations might have neutral or even beneficial effects under certain conditions, many causative mutations are associated with detrimental outcomes. They can lead to various genetic disorders, diseases, or functional impairments in organisms by disrupting normal biological processes or cellular functions.</p>
<section id="beneficial-mutations" class="level3" data-number="11.5.1">
<h3 data-number="11.5.1" class="anchored" data-anchor-id="beneficial-mutations"><span class="header-section-number">11.5.1</span> Beneficial Mutations</h3>
<p><strong>Beneficial mutations</strong> are genetic changes that confer advantages to an organism, increasing its fitness or survival in a particular environment. Unlike deleterious mutations that decrease fitness, beneficial mutations lead to positive selection, causing their allele frequencies to rise rapidly over successive generations within a population.</p>
<p>Imagine if “LAZY” were a beneficial mutation instead of a slightly deleterious one. In this case, individuals carrying the beneficial mutation would have a higher chance of surviving, reproducing, or thriving in their environment compared to others without the mutation. This advantage would lead to the rapid increase in the frequency of the mutated allele in the population.</p>
<p>As this advantageous allele becomes more prevalent, it undergoes a process known as <strong>genomic hitchhiking</strong>. Alleles that are physically close to the beneficial mutation on the chromosome also experience an increase in frequency because they are in linkage disequilibrium with the advantageous allele. Linkage disequilibrium refers to the non-random association of alleles at different loci.</p>
<p>The phenomenon of genomic sweeps occurs as a result of this process. Genomic sweeps involve regions in the genome with fewer genetic variations or Single Nucleotide Polymorphisms (SNPs) due to the fixation of alleles associated with the beneficial mutation. Essentially, all the alleles in that region have reached fixation, resulting in reduced genetic diversity in those specific genomic regions across the population.</p>
</section>
<section id="sweep-analysis" class="level3" data-number="11.5.2">
<h3 data-number="11.5.2" class="anchored" data-anchor-id="sweep-analysis"><span class="header-section-number">11.5.2</span> Sweep Analysis</h3>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../imgs/week11/sweep.png" class="img-fluid figure-img" width="382"></p>
<p></p><figcaption class="figure-caption">Sweep Analysis</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>During a selective sweep, a beneficial mutation increases in frequency and leads to a reduction in genetic diversity in the surrounding genomic area. One consequence is a decrease in the number of intermediate-frequency alleles in the region affected by the sweep.</p>
<p>Linked genetic variations, particularly those associated with the beneficial allele, experience an increase in frequency due to the selective sweep. These intermediate alleles linked to the advantageous mutation tend to shift toward higher frequencies, transforming into high-frequency Single Nucleotide Polymorphisms (SNPs) within the affected region.</p>
<p>As the sweep progresses and the advantageous allele reaches fixation or near-fixation in the population, a new scenario emerges. Following the sweep, new mutations begin to accumulate in the region previously impacted by the selective sweep.</p>
<section id="one-possible-approach" class="level4" data-number="11.5.2.1">
<h4 data-number="11.5.2.1" class="anchored" data-anchor-id="one-possible-approach"><span class="header-section-number">11.5.2.1</span> One Possible Approach</h4>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../imgs/week11/method1.png" class="img-fluid figure-img" width="148"></p>
<p></p><figcaption class="figure-caption">One Method for Sweep Analysis</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>This method involves assessing the frequency of derived alleles—genetic variants that differ from the ancestral form—near the SNP subjected to selection. When a beneficial mutation undergoes a selective sweep, a notable increase in high-frequency derived alleles is observed, signifying that these SNPs have reached fixation in the population due to positive selection.</p>
<p>Conversely, the presence of low-frequency derived alleles indicates the emergence of new genetic variations subsequent to the selective sweep. During this process, the number of intermediate-frequency alleles typically decreases due to the impact of the sweep.</p>
<p>To perform this analysis, researchers calculate the site frequency spectrum (SFS) locally around the SNP of interest. The SFS represents the distribution of allele frequencies in a specific genomic region. By comparing this local SFS to the average SFS across the entire genome, scientists can identify deviations or shifts in allele frequencies caused by the selective sweep.</p>


</section>
</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  let localAlternateSentinel = 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } 
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../chapters/week10.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">Re-Sequencing and SNP Calling</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../chapters/week12.html" class="pagination-link">
        <span class="nav-page-text"><span class="chapter-number">12</span>&nbsp; <span class="chapter-title">Comparative Genomics</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->



</body></html>