<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.1.189">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>BS4017: High Throughput Bioinformatics - 7&nbsp; Transcriptomes: Differential Expression to Enrichment Analyses</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../chapters/week8.html" rel="next">
<link href="../chapters/week6.html" rel="prev">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../site_libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>
<script async="" src="https://hypothes.is/embed.js"></script>

  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
    <div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Transcriptomes: Differential Expression to Enrichment Analyses</span></h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../">BS4017: High Throughput Bioinformatics</a> 
        <div class="sidebar-tools-main tools-wide">
    <a href="https://github.com/kevfo/BS4017_Notes" title="Source Code" class="sidebar-tool px-1"><i class="bi bi-github"></i></a>
    <a href="" title="Share" id="sidebar-tool-dropdown-0" class="sidebar-tool dropdown-toggle px-1" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi bi-share"></i></a>
    <ul class="dropdown-menu" aria-labelledby="sidebar-tool-dropdown-0">
        <li>
          <a class="dropdown-item sidebar-tools-main-item" href="https://twitter.com/intent/tweet?url=|url|">
            <i class="bi bi-bi-twitter pe-1"></i>
          Twitter
          </a>
        </li>
        <li>
          <a class="dropdown-item sidebar-tools-main-item" href="https://www.facebook.com/sharer/sharer.php?u=|url|">
            <i class="bi bi-bi-facebook pe-1"></i>
          Facebook
          </a>
        </li>
    </ul>
  <a href="" class="quarto-color-scheme-toggle sidebar-tool" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
  <a href="" class="quarto-reader-toggle sidebar-tool" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
    </div>
      </div>
      <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link">Preface</a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true"><strong>PART I: LECTURE NOTES</strong></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/week1.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Introduction to Linux and the Command Line (untested)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/week2.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Genomic Sequencing and Databases</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/week3.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Data Preprocessing and Quality Control</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/week4.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title"><em>De Novo</em> Sequencing</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/week5.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Annotation</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/week6.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Transcriptomes: Experiment Design and RNA Sequence Data Processing</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/week7.html" class="sidebar-item-text sidebar-link active"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Transcriptomes: Differential Expression to Enrichment Analyses</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/week8.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Transcriptomes: Network Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/week9.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">SNPs, Structural Variations, and Pan-Genomes</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/week10.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">Re-Sequencing and SNP Calling</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/week11.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">11</span>&nbsp; <span class="chapter-title">Transcriptomes: Network Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/week12.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">12</span>&nbsp; <span class="chapter-title">Transcriptomes: Network Analysis</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#fundamental-metrics" id="toc-fundamental-metrics" class="nav-link active" data-scroll-target="#fundamental-metrics"><span class="toc-section-number">7.1</span>  Fundamental Metrics</a>
  <ul class="collapse">
  <li><a href="#fold-change" id="toc-fold-change" class="nav-link" data-scroll-target="#fold-change"><span class="toc-section-number">7.1.1</span>  Fold Change</a></li>
  <li><a href="#ma-plots" id="toc-ma-plots" class="nav-link" data-scroll-target="#ma-plots"><span class="toc-section-number">7.1.2</span>  MA Plots</a></li>
  </ul></li>
  <li><a href="#testing-significance" id="toc-testing-significance" class="nav-link" data-scroll-target="#testing-significance"><span class="toc-section-number">7.2</span>  Testing Significance</a>
  <ul class="collapse">
  <li><a href="#p-values" id="toc-p-values" class="nav-link" data-scroll-target="#p-values"><span class="toc-section-number">7.2.1</span>  P Values</a></li>
  <li><a href="#differences-between-conditions" id="toc-differences-between-conditions" class="nav-link" data-scroll-target="#differences-between-conditions"><span class="toc-section-number">7.2.2</span>  Differences Between Conditions</a></li>
  <li><a href="#models-to-analyze-rna-sequencing-data" id="toc-models-to-analyze-rna-sequencing-data" class="nav-link" data-scroll-target="#models-to-analyze-rna-sequencing-data"><span class="toc-section-number">7.2.3</span>  Models to Analyze RNA Sequencing Data</a></li>
  <li><a href="#biological-vs.-statistical-significance" id="toc-biological-vs.-statistical-significance" class="nav-link" data-scroll-target="#biological-vs.-statistical-significance"><span class="toc-section-number">7.2.4</span>  Biological vs.&nbsp;Statistical Significance</a></li>
  <li><a href="#multiple-testing" id="toc-multiple-testing" class="nav-link" data-scroll-target="#multiple-testing"><span class="toc-section-number">7.2.5</span>  Multiple Testing</a></li>
  <li><a href="#classical-correction-methods" id="toc-classical-correction-methods" class="nav-link" data-scroll-target="#classical-correction-methods"><span class="toc-section-number">7.2.6</span>  Classical Correction Methods</a></li>
  <li><a href="#false-discovery-rate" id="toc-false-discovery-rate" class="nav-link" data-scroll-target="#false-discovery-rate"><span class="toc-section-number">7.2.7</span>  False Discovery Rate</a></li>
  </ul></li>
  <li><a href="#differentially-expressed-gene-and-biology" id="toc-differentially-expressed-gene-and-biology" class="nav-link" data-scroll-target="#differentially-expressed-gene-and-biology"><span class="toc-section-number">7.3</span>  Differentially Expressed Gene and Biology</a>
  <ul class="collapse">
  <li><a href="#tool-recommendations" id="toc-tool-recommendations" class="nav-link" data-scroll-target="#tool-recommendations"><span class="toc-section-number">7.3.1</span>  Tool Recommendations</a></li>
  </ul></li>
  <li><a href="#analysis-outcomes" id="toc-analysis-outcomes" class="nav-link" data-scroll-target="#analysis-outcomes"><span class="toc-section-number">7.4</span>  Analysis Outcomes</a>
  <ul class="collapse">
  <li><a href="#levels-of-analysis" id="toc-levels-of-analysis" class="nav-link" data-scroll-target="#levels-of-analysis"><span class="toc-section-number">7.4.1</span>  Levels of Analysis</a></li>
  <li><a href="#gene-ontology-i.e.-go" id="toc-gene-ontology-i.e.-go" class="nav-link" data-scroll-target="#gene-ontology-i.e.-go"><span class="toc-section-number">7.4.2</span>  Gene Ontology (i.e., GO)</a></li>
  <li><a href="#how-does-gene-set-analysis-work" id="toc-how-does-gene-set-analysis-work" class="nav-link" data-scroll-target="#how-does-gene-set-analysis-work"><span class="toc-section-number">7.4.3</span>  How Does Gene Set Analysis Work?</a></li>
  </ul></li>
  <li><a href="#enrichment-analysis" id="toc-enrichment-analysis" class="nav-link" data-scroll-target="#enrichment-analysis"><span class="toc-section-number">7.5</span>  Enrichment Analysis</a>
  <ul class="collapse">
  <li><a href="#singular-enrichment-analysis" id="toc-singular-enrichment-analysis" class="nav-link" data-scroll-target="#singular-enrichment-analysis"><span class="toc-section-number">7.5.1</span>  Singular Enrichment Analysis</a></li>
  </ul></li>
  <li><a href="#gene-set-enrichment" id="toc-gene-set-enrichment" class="nav-link" data-scroll-target="#gene-set-enrichment"><span class="toc-section-number">7.6</span>  Gene Set Enrichment</a>
  <ul class="collapse">
  <li><a href="#gene-set-enrichment-analysis" id="toc-gene-set-enrichment-analysis" class="nav-link" data-scroll-target="#gene-set-enrichment-analysis"><span class="toc-section-number">7.6.1</span>  Gene Set Enrichment Analysis</a></li>
  <li><a href="#gene-set-analyses" id="toc-gene-set-analyses" class="nav-link" data-scroll-target="#gene-set-analyses"><span class="toc-section-number">7.6.2</span>  Gene Set Analyses</a></li>
  </ul></li>
  </ul>
<div class="toc-actions"><div><i class="bi bi-github"></i></div><div class="action-links"><p><a href="https://github.com/kevfo/BS4017_Notes/edit/main/chapters/week7.qmd" class="toc-action">Edit this page</a></p><p><a href="https://github.com/kevfo/BS4017_Notes/issues/new" class="toc-action">Report an issue</a></p></div></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title d-none d-lg-block"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Transcriptomes: Differential Expression to Enrichment Analyses</span></h1>
</div>



<div class="quarto-title-meta">

    
    
  </div>
  

</header>

<p>Let’s explore how we go from raw read counts to identifying differential gene expression in a straightforward way.</p>
<p>Imagine you have a collection of different-sized baskets, each containing different numbers of apples (gene counts). Your goal is to figure out which baskets have more or fewer apples, but you need a consistent way to do this.</p>
<p>Now, in the world of gene expression analysis, different software tools come with their own methods for counting genes, but they typically start with unnormalized count data, just like having apples in baskets without any adjustments.</p>
<p>In your analysis, when you use a method like pseudoalignment, it not only provides raw counts (the number of apples in each basket) but also normalized values called TPM (Transcripts per Million), which are like having apples per million baskets.</p>
<p>However, when it comes to finding genes that are differentially expressed (meaning they have significantly more or fewer apples), we usually work with the raw count data, not TPM values. It’s like counting apples directly rather than adjusting for the number of baskets.</p>
<p>Here’s a standard workflow:</p>
<ol type="1">
<li><p><strong>Read in Raw Count Data</strong></p>
<p>You start by bringing in the raw count data (the number of apples in each basket) into a data analysis tool like R.</p></li>
<li><p><strong>Normalization and Modeling</strong></p>
<p>Then, you apply software-specific normalization methods to make the data more suitable for analysis. You might also use a (generalized) linear model, which is like a mathematical tool, to calculate fold changes (how much more or less) for each gene and their associated p-values (a measure of significance).</p></li>
</ol>
<p>In essence, you’re like a fruit analyst, taking the raw count data, adjusting it if needed, and using statistical models to figure out which baskets have significantly different numbers of apples. This is how we identify differential gene expression, helping us understand which genes are playing a more significant role in specific conditions or situations.</p>
<section id="fundamental-metrics" class="level2" data-number="7.1">
<h2 data-number="7.1" class="anchored" data-anchor-id="fundamental-metrics"><span class="header-section-number">7.1</span> Fundamental Metrics</h2>
<section id="fold-change" class="level3" data-number="7.1.1">
<h3 data-number="7.1.1" class="anchored" data-anchor-id="fold-change"><span class="header-section-number">7.1.1</span> Fold Change</h3>
<p>Let’s discuss the concept of “Fold Change” and how it serves as a measure of differential gene expression.</p>
<p>Imagine you have two baskets of apples: one from a case scenario (like a treatment group) and the other from a control scenario (like a standard group). You want to know if the number of apples in the case basket is different from the control basket. Fold change is a simple way to express this difference.</p>
<p>Here’s how it works:</p>
<ol type="1">
<li><p><strong>Calculating Fold Change</strong></p>
<p><span class="math display">\[\begin{equation}
  \log\left(\frac{\text{Case}}{\text{Control}}\right) = \log\left(\text{Case}\right) - \log\left(\text{Control}\right)
\end{equation}\]</span></p>
<p>To calculate fold change, you take the number of apples (gene counts) in the case basket and divide it by the number of apples in the control basket. This ratio tells you how many times more (or less) apples there are in the case compared to the control.</p></li>
<li><p><strong>Log2 Transformation</strong></p>
<p>Fold changes are often reported in the log2 domain because it has some useful properties. When you take the logarithm of the fold change, it simplifies the interpretation. Also, it’s symmetric, meaning that if a gene is upregulated (more apples in the case), it will have a positive value, and if it’s downregulated (fewer apples in the case), it will have a negative value.</p></li>
<li><p><strong>Relationship to Difference</strong></p>
<p>The <span class="math inline">\(\log2\)</span> of the fold change is related to the difference between the case and control. It’s like expressing the change in the number of apples as a difference in their logarithms.</p></li>
<li><p><strong>Geometric Mean</strong></p>
<p><span class="math display">\[\begin{equation}
  \log\left(\sqrt{\text{Case} \cdot \text{Control}}\right) = \frac{\log\left(\text{Case}\right) + \log\left(\text{Control}\right)}{2}
\end{equation}\]</span></p>
<p>Another way to express fold change is to use the geometric mean of the apples in the case and control baskets. This is like finding the average number of apples between the two baskets, but in the <span class="math inline">\(\log2\)</span> domain.</p></li>
</ol>
<p>However, fold change has limitations. It works well when you have only one measurement, but it’s not ideal when your data is very noisy or when you’re dealing with complex datasets. Noise can lead to misleading results, which is why more sophisticated statistical methods are often used for differential gene expression analysis in RNA-seq data.</p>
</section>
<section id="ma-plots" class="level3" data-number="7.1.2">
<h3 data-number="7.1.2" class="anchored" data-anchor-id="ma-plots"><span class="header-section-number">7.1.2</span> MA Plots</h3>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../imgs/week7/ma_plot.png" class="img-fluid figure-img" width="510"></p>
<p></p><figcaption class="figure-caption">Examples of MA Plots</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>Imagine you have data points on a graph, where one axis represents the log fold change (how much something has changed on a logarithmic scale, like we discussed earlier), and the other axis represents the mean expression (the average level of something, like the average number of apples in our baskets).</p>
<p>Now, when you look at this scatterplot of log fold change, it might be a bit hard to interpret because it’s not as intuitive as seeing the actual fold change values directly. The MA plot comes to the rescue.</p>
<p>Here’s how it works:</p>
<ol type="1">
<li><p><strong>Transforming the Axes</strong></p>
<p>The MA plot essentially takes the scatterplot and rotates the axes by 45 degrees. This means that the log fold change values are now on one axis (the M-axis), and the mean expression values are on the other axis (the A-axis).</p></li>
<li><p><strong>M-A Axes</strong></p>
<p>The M-axis typically represents the difference between the log fold change in the case and control groups, while the A-axis represents the average of the log fold change values.</p></li>
<li><p><strong>Interpretable Plot</strong></p>
<p>Now, instead of looking at log fold change values, you’re looking at something more intuitive – the actual fold change values (like we discussed earlier) on one axis, and the mean expression levels on the other axis. This makes it easier to see how much things have changed in terms of fold change, relative to the average expression.</p></li>
</ol>
</section>
</section>
<section id="testing-significance" class="level2" data-number="7.2">
<h2 data-number="7.2" class="anchored" data-anchor-id="testing-significance"><span class="header-section-number">7.2</span> Testing Significance</h2>
<p>Imagine you have two groups of apples, like a treatment group and a control group. You want to know if there’s a real difference in the number of apples between these two groups or if any difference could have occurred by chance.</p>
<p>Here’s how significance testing works:</p>
<ol type="1">
<li><p><strong>Construct a Null Hypothesis</strong></p>
<p>This is like making an educated guess based on the idea that there’s no real difference between the two groups. In a two-sample case (like treatment vs.&nbsp;control), the null hypothesis would be something like “the mean number of apples in the treatment group is the same as the mean number of apples in the control group.” In a one-sample case, it might be “the mean number of apples is not different from zero.”</p></li>
<li><p><strong>Test the Hypothesis</strong></p>
<p>You perform statistical tests to see if this null hypothesis holds. The tests rely on certain assumptions about the data, such as following a known distribution (e.g., Student’s t-distribution for comparing means) and having a test statistic with a known distribution under the null hypothesis.</p></li>
<li><p><strong>P-value</strong></p>
<p>After running the statistical test, you get a p-value. This p-value tells you the probability of obtaining the data you have (or more extreme data) by chance alone, assuming that the null hypothesis is true. In simple terms, it’s like asking, “How likely is it that the observed difference in apples between the two groups happened purely by random chance?”</p></li>
<li><p><strong>Interpreting the P-value</strong></p>
<p>If the p-value is small, typically less than 0.01 or 0.05 (but this threshold can vary), it suggests that the difference you observed is unlikely to have occurred purely by chance. In other words, it’s statistically significant. This means you have evidence to reject the null hypothesis and conclude that there’s likely a real difference between the groups.</p></li>
</ol>
<section id="p-values" class="level3" data-number="7.2.1">
<h3 data-number="7.2.1" class="anchored" data-anchor-id="p-values"><span class="header-section-number">7.2.1</span> P Values</h3>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../imgs/week7/p-values.png" class="img-fluid figure-img" width="254"></p>
<p></p><figcaption class="figure-caption">Illustration of a P-Value</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>Imagine you’re playing a game with a coin. You want to know if the coin is fair or biased, meaning it might not land on heads and tails with equal probability. To test this, you flip the coin a bunch of times and record the outcomes.</p>
<p>Now, the null hypothesis here is that the coin is fair. In other words, there’s no bias; it’s just like any other regular coin. The alternative hypothesis is that the coin is biased.</p>
<p>Here’s how p-values come into play:</p>
<ol type="1">
<li><p><strong>Calculating the p-value</strong></p>
<p>When you calculate a p-value, you’re essentially figuring out the probability that you would get the results you did (like getting heads or tails a certain number of times) if the null hypothesis were true. In other words, it’s like asking, “What’s the chance that the coin appears biased purely by coincidence?”</p></li>
<li><p><strong>Interpreting the p-value</strong></p>
<p>If the p-value is small (typically less than 0.01 or 0.05), it suggests that the results you observed are unlikely to have occurred purely by chance. So, if you find a very small p-value, you might conclude that there’s evidence against the null hypothesis (the coin is fair), and you start to consider the alternative hypothesis (the coin is biased).</p></li>
</ol>
<p>In gene expression analysis, p-values work similarly. They tell you the chance that a gene is categorized as differentially expressed (meaning its expression levels have changed significantly) purely by coincidence or random variation.</p>
<p>For example, if you’re comparing gene expression in two groups, a small p-value suggests that the difference you observed in gene expression is unlikely to have happened by random chance alone. It provides evidence that the gene is genuinely differentially expressed between the groups.</p>
</section>
<section id="differences-between-conditions" class="level3" data-number="7.2.2">
<h3 data-number="7.2.2" class="anchored" data-anchor-id="differences-between-conditions"><span class="header-section-number">7.2.2</span> Differences Between Conditions</h3>
<p>Let’s explore two common statistical tests used to determine if there is a significant difference between two conditions in gene expression analysis, the Student’s t-test and the Wilcoxon Rank Sum (or Wilcoxon Mann-Whitney) test.</p>
<p>Imagine you have data from two groups, like a treatment group and a control group, and you want to know if there’s a real difference between them.</p>
<ol type="1">
<li><p><strong>Student’s t-test:</strong></p>
<ol type="1">
<li><p><strong>Testing Means:</strong> The Student’s t-test is like a tool for comparing the means (averages) of the two groups. It tells you if the difference in means between the groups is statistically significant.</p></li>
<li><p><strong>Types of t-tests:</strong> There are various versions of the t-test, like the two-sample t-test (comparing two groups), one-sample t-test (comparing one group to a known value), and variations that account for assumptions like equal or unequal variances between the groups.</p></li>
<li><p><strong>Assumption:</strong> The t-test assumes that the data is normally distributed, meaning it follows a typical bell-shaped curve around the mean. This assumption is important because the test relies on normal distribution properties.</p></li>
</ol></li>
<li><p><strong>Wilcoxon Rank Sum (or Wilcoxon Mann-Whitney) test:</strong></p>
<ol type="1">
<li><p><strong>Testing Medians:</strong> The Wilcoxon Rank Sum test, also known as the Mann-Whitney test, is different from the t-test. It compares the location, specifically the medians, of the two groups. It tells you if the medians are significantly different.</p></li>
<li><p><strong>Types of Wilcoxon tests:</strong> Like the t-test, you can use variations of the Wilcoxon test, including two-sample Wilcoxon (comparing two groups) and one-sample Wilcoxon (comparing one group to a known value).</p></li>
<li><p><strong>Assumption:</strong> The Wilcoxon test doesn’t rely on the assumption of normal distribution. Instead, it works well when the distributions are symmetric and similar.</p></li>
<li><p><strong>Sensitivity:</strong> The Wilcoxon test is often considered less powerful (meaning it’s less likely to detect a true difference) compared to the t-test when the data is close to a normal distribution. However, it’s more robust when dealing with data that doesn’t meet the normal distribution assumption.</p></li>
</ol></li>
</ol>
<p>In summary, both the Student’s t-test and the Wilcoxon Rank Sum test are used to determine if there’s a significant difference between two conditions. The t-test focuses on means and assumes normal distribution, while the Wilcoxon test examines medians and doesn’t rely on normal distribution assumptions. The choice between them depends on your data and whether the assumptions of the t-test are met.</p>
</section>
<section id="models-to-analyze-rna-sequencing-data" class="level3" data-number="7.2.3">
<h3 data-number="7.2.3" class="anchored" data-anchor-id="models-to-analyze-rna-sequencing-data"><span class="header-section-number">7.2.3</span> Models to Analyze RNA Sequencing Data</h3>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../imgs/week7/models.png" class="img-fluid figure-img" width="486"></p>
<p></p><figcaption class="figure-caption">Model Transformation for RNA Sequencing Data</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>When we’re working with RNA sequencing data, we use something called <strong>generalized linear models</strong> to transform our data to follow a <strong>Gaussian distribution</strong> (i.e., a bell curve) - especially error values in transcription to positive values.</p>
<p>Such a transformation often takes into account the fact that the data does not follow a Gaussian distribution. Several tools have different ways of doing this:</p>
<ol type="1">
<li><p><strong>Using theory</strong></p>
<p>Tools like EdgeR and DESeq2 (i.e., from BioConductor in R) use the Negative Binomial distribution or the Poisson distribution (e.g., PoissonSeq does this).</p></li>
</ol>
</section>
<section id="biological-vs.-statistical-significance" class="level3" data-number="7.2.4">
<h3 data-number="7.2.4" class="anchored" data-anchor-id="biological-vs.-statistical-significance"><span class="header-section-number">7.2.4</span> Biological vs.&nbsp;Statistical Significance</h3>
<p>When we talk about biological significance and statistical significance, we’re trying to figure out if something we see in our data is not only mathematically significant but also meaningful in the real world of biology. You see, sometimes the numbers we get from our experiments might be statistically significant, which means they’re different enough to catch our attention, but they might be so tiny that they don’t actually matter in the bigger picture of how a living thing functions.</p>
<p>To decide if something is genuinely important in biology, we need to consider two things. First, we look at the average change or difference we observe, and second, we check if this change is statistically significant. Think of it like this: we’re not just interested in whether something is different; we also want to know if that difference is big enough to make a real impact.</p>
<p>To help us figure this out, scientists often use a rule of thumb: if the change is at least ±1 log2-ratio, it’s usually considered biologically meaningful. But here’s the tricky part: this “meaningful” threshold can vary depending on what kind of cells we’re studying (like brain cells versus liver cells) and what we’re investigating in our experiment (like subtle changes in how a gene works versus genes that react really strongly to something).</p>
</section>
<section id="multiple-testing" class="level3" data-number="7.2.5">
<h3 data-number="7.2.5" class="anchored" data-anchor-id="multiple-testing"><span class="header-section-number">7.2.5</span> Multiple Testing</h3>
<p>The thing about high throughput bioinformatics is that we usually need to have a lot of tests to measure significances between different things. However, the problem with such tests is that genes that are said to be significantly expressed may actually be false positives in disguise.</p>
</section>
<section id="classical-correction-methods" class="level3" data-number="7.2.6">
<h3 data-number="7.2.6" class="anchored" data-anchor-id="classical-correction-methods"><span class="header-section-number">7.2.6</span> Classical Correction Methods</h3>
<p>Imagine you’re in charge of testing a bunch of different things, like trying to figure out which light switches in a giant building work. If you just randomly flip switches and say, “This one works!” you might make mistakes. So, you set a rule: if you’re going to say a switch works, you want to be pretty darn sure it’s working, to avoid false alarms.</p>
<p>In the world of statistics, when we do lots of tests (like checking the switches), we want to make sure we’re not saying things are true when they’re not. That’s where these correction methods come in:</p>
<ol type="1">
<li><p><strong>Sidak Correction and Bonferroni Correction</strong></p>
<p><span class="math display">\[\begin{align}
  \text{Sidak } \alpha_e &amp;= 1 - \sqrt[m]{1 - \alpha_e} \\
  \text{Bonferroni } \alpha_e &amp;= \frac{\alpha_e}{m}
\end{align}\]</span></p>
<p>These are like rules we use to decide how confident we need to be before we say something is true (like a switch works). The idea is to set a threshold for each test. If the chance of being wrong (a false positive) is too high, we won’t call it true.</p></li>
<li><p><strong>Family-wise Error Rate</strong></p>
<p>This is just a fancy way of saying that we’re looking at the overall rate of false positives when we do many tests. We don’t want too many switches to be labeled as working when they’re not.</p></li>
</ol>
<p>Now, here’s the catch: these methods are sometimes seen as overly cautious. Why? Well, first, the tests we do aren’t always completely independent, meaning the results of one test might affect the next one. Second, they control for something a bit different: the chance that all the tests are correct, not just one. So, sometimes they might make us miss switches that actually work because we’re being too strict.</p>
</section>
<section id="false-discovery-rate" class="level3" data-number="7.2.7">
<h3 data-number="7.2.7" class="anchored" data-anchor-id="false-discovery-rate"><span class="header-section-number">7.2.7</span> False Discovery Rate</h3>
<p>Imagine you’re a detective trying to solve a mystery. In this mystery, you’re looking for clues in different places, and you want to make sure you catch as many real clues as possible, but you’re okay with the idea that you might pick up a few fake clues along the way.</p>
<p>Now, in the world of statistics, when we’re doing lots of tests or looking for lots of clues, we have to decide how strict we want to be about avoiding false leads (false positives).</p>
<p>Traditionally, we used to focus on controlling the <strong>family-wise error rate</strong> (i.e., <strong>FWER</strong>), which meant we wanted to make sure we didn’t make any mistakes anywhere in our tests. It’s like saying, “I don’t want a single false clue in my investigation.”</p>
<p><span class="math display">\[\begin{equation}
  \text{\#False Discovery Rate} = \frac{\text{\#False Positives}}{\text{\#True Positives + \#False Positives}}
\end{equation}\]</span></p>
<p>But sometimes, being too strict about this can make us miss out on real discoveries. So, instead of worrying too much about catching every single error, we can control the <strong>false discovery rate</strong> (i.e., <strong>FDR</strong>). This approach allows us to be a bit more flexible. It means that we’re okay with accepting a certain proportion of our significant findings as potentially false positives.</p>
<section id="benjamini-hochberg-correction" class="level4" data-number="7.2.7.1">
<h4 data-number="7.2.7.1" class="anchored" data-anchor-id="benjamini-hochberg-correction"><span class="header-section-number">7.2.7.1</span> Benjamini-Hochberg Correction</h4>
<p>Imagine you’re a scientist working with a lot of data, like testing many genes in an experiment. You want to make sure that when you say a gene is significant, it’s not just a random chance finding. The Benjamini-Hochberg correction helps you do just that.</p>
<p>Here’s how it works:</p>
<ol type="1">
<li><p><strong>Choose Your Significance Level:</strong> First, you decide how strict you want to be. This is like setting a rule that says, “I want to be this confident before I call something significant.” For example, you might choose a 5% significance level, which means you’re willing to accept a 5% chance of making a mistake.</p></li>
<li><p><strong>Compute Many Individual Tests:</strong> Next, you run lots of individual tests, one for each gene you’re studying. These tests tell you how likely it is that each gene’s results are just random.</p></li>
<li><p><strong>Sort the p-values:</strong> Now, you have a bunch of p-values, which are like scores that tell you how likely it is that something is significant. You sort these p-values from smallest to largest, like arranging them in order.</p></li>
<li><p><strong>Compute Gene-Specific Thresholds:</strong> Here’s where the Benjamini-Hochberg correction comes in. You calculate a specific threshold for each gene based on its p-value and the significance level you chose in step 1. This threshold helps you decide which genes are significant.</p></li>
</ol>
<p>Here’s what’s cool about this method: It’s a bit like being a careful scientist who wants to be sure about their findings. It doesn’t just apply the same rule to all genes; it customizes the rules for each gene based on its own data. So, some genes might have to meet a higher bar to be called significant, while others have a lower bar, depending on their p-values and the chosen significance level.</p>
<p>In simple terms, the Benjamini-Hochberg correction helps you avoid calling something significant when it’s not, while also being flexible enough to account for the individual uniqueness of each gene in your study. It’s like being both cautious and fair in your analysis of a big dataset.</p>
</section>
<section id="q-values" class="level4" data-number="7.2.7.2">
<h4 data-number="7.2.7.2" class="anchored" data-anchor-id="q-values"><span class="header-section-number">7.2.7.2</span> Q Values</h4>
<p>Imagine you’re exploring a big forest looking for hidden treasures, but you’re also worried about accidentally finding something that isn’t a treasure.</p>
<p>Now, when scientists like to discover things in their experiments, they’re also concerned about making mistakes, like saying they found something important when it’s not actually there. That’s where q-values come in.</p>
<ol type="1">
<li><p><strong>Estimating True Negatives:</strong> Q-values help scientists figure out how many times they might have made mistakes by calling something significant when it’s not. You can think of it as estimating the number of times they thought they found a treasure, but it turned out to be just a rock. This is something that the Benjamini-Hochberg method doesn’t really focus on.</p></li>
<li><p><strong>Q-values vs.&nbsp;P-values:</strong> Q-values are a bit like p-values, which you can think of as scores for how likely something is to be important. But q-values go a step further. They take a bunch of these p-values and put them in order. Then, they help scientists understand how many of these “important” things might actually be mistakes.</p></li>
</ol>
</section>
</section>
</section>
<section id="differentially-expressed-gene-and-biology" class="level2" data-number="7.3">
<h2 data-number="7.3" class="anchored" data-anchor-id="differentially-expressed-gene-and-biology"><span class="header-section-number">7.3</span> Differentially Expressed Gene and Biology</h2>
<section id="tool-recommendations" class="level3" data-number="7.3.1">
<h3 data-number="7.3.1" class="anchored" data-anchor-id="tool-recommendations"><span class="header-section-number">7.3.1</span> Tool Recommendations</h3>
<section id="sahraeian-et.-al" class="level4" data-number="7.3.1.1">
<h4 data-number="7.3.1.1" class="anchored" data-anchor-id="sahraeian-et.-al"><span class="header-section-number">7.3.1.1</span> Sahraeian et. al</h4>
<p>When it comes to analyzing RNA-seq data, there are some recommendations from a guy called Sahraeian and his team that can help us understand which tools work well. Imagine you have a toolbox, and you want to know which tools are the best for the job.</p>
<ol type="1">
<li><p><strong>Quantification</strong></p>
<p>One of the things they looked at was how well different tools work to measure gene expression. They found that tools like HISAT2 and TopHat are better at this job compared to STAR. Think of it like some wrenches being better at turning bolts than others.</p></li>
<li><p><strong>Efficiency Matters</strong></p>
<p>They also found that some tools that don’t rely on traditional alignment methods (alignment-free tools) are quite efficient. However, when they combined a tool called StringTie with an efficient aligner like HISAT2, it turned out to be the most efficient approach. So, it’s like saying, sometimes it’s better to use two tools together to get the job done quickly.</p></li>
<li><p><strong>Speed Differences</strong></p>
<p>Some tools were much faster than others, but they didn’t provide specific numbers in this study. Think of it as some tools being super-fast like a race car, while others are a bit slower like regular cars, but they didn’t give us the exact speed limits.</p></li>
<li><p><strong>Differential Expression</strong></p>
<p>Another important task in RNA-seq analysis is figuring out which genes are different between conditions. They found that a tool called edgeR did a great job at this and had a lower rate of falsely calling genes as different when they weren’t.</p></li>
<li><p><strong>Best Recommendation</strong></p>
<p>Overall, their top recommendation for differential expression analysis was a tool called DESeq2. It’s like saying, if you want to pick just one tool from your toolbox, DESeq2 might be the best all-around choice. But remember, the best tool can vary depending on the specific job or data you’re working with.</p></li>
</ol>
</section>
<section id="corchete-et.-al" class="level4" data-number="7.3.1.2">
<h4 data-number="7.3.1.2" class="anchored" data-anchor-id="corchete-et.-al"><span class="header-section-number">7.3.1.2</span> Corchete et. al</h4>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../imgs/week7/corchete.png" class="img-fluid figure-img" width="236"></p>
<p></p><figcaption class="figure-caption">Corchete et. al’s Work</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>In this study by Corchete and their team, they explored various ways to process and analyze RNA-seq data to find differentially expressed genes. Think of it like they tried out many recipes to cook the same dish and wanted to know which one tasted the best.</p>
<ol type="1">
<li><p><strong>Exploring Different Combinations</strong></p>
<p>The researchers didn’t just stick to one way of analyzing the data; they tried a whopping 192 different methods or combinations of steps to see how they affected the results. Imagine if you were making a sandwich, and you tried using different types of bread, spreads, and fillings to find the most delicious combination.</p></li>
<li><p><strong>Samples and Validation</strong></p>
<p>To make sure their findings were reliable, they used 18 samples from two different human cell lines. It’s like cooking the same meal multiple times to make sure you get consistent results. They also validated their findings by using a technique called qRT-PCR on the same samples, which is a bit like double-checking the taste of your dish to be absolutely sure it’s delicious.</p></li>
<li><p><strong>Recommended Preprocessing Steps</strong></p>
<p>After trying all these different recipes (or methods), they found that a combination of preprocessing steps worked well. It included tools like Trimmomatic, RUM, HTSeq Union, and TMM. Think of it as using specific ingredients and techniques in a recipe to make your dish turn out just right.</p></li>
<li><p><strong>The Limma Trend</strong></p>
<p>Among all the methods they tested, one called Limma trend consistently performed well. It’s like finding that a particular way of seasoning your dish always makes it taste great.</p></li>
</ol>
</section>
</section>
</section>
<section id="analysis-outcomes" class="level2" data-number="7.4">
<h2 data-number="7.4" class="anchored" data-anchor-id="analysis-outcomes"><span class="header-section-number">7.4</span> Analysis Outcomes</h2>
<p>There are usually tens of, if not hundreds or even thousands of differentially expressed genes.</p>
<p>When we’re working with RNA-seq data and trying to figure out which genes are significant, it’s like solving a puzzle. We have lots of genes, and for each one, we use a special test to check if it’s behaving differently under different experimental conditions. We also make sure our results are reliable by correcting the test results. What we end up with is a list of genes that seem to be important.</p>
<p>But here’s the thing: just having this list of genes might not give us the whole picture. It’s a bit like having puzzle pieces without knowing what the final picture looks like. And depending on the tool we use to analyze the data, the list of important genes can vary.</p>
<p>Now, why are we doing all this in the first place? The main goal of RNA-seq is to understand how things change between different experimental conditions. We want to know what’s happening at a functional or molecular level. It’s like trying to figure out what’s going on inside a black box.</p>
<section id="levels-of-analysis" class="level3" data-number="7.4.1">
<h3 data-number="7.4.1" class="anchored" data-anchor-id="levels-of-analysis"><span class="header-section-number">7.4.1</span> Levels of Analysis</h3>
<p>When scientists are studying gene behavior, they look at things on different levels, a bit like zooming in and out with a microscope. It’s kind of like exploring a forest where you can start with the smallest details and gradually uncover the bigger picture.</p>
<ol type="1">
<li><p><strong>Single Gene Level:</strong> At the smallest level, researchers focus on single genes. They want to know if a particular gene behaves differently when it’s exposed to a treatment or experiment compared to when it’s in its normal state (the control). It’s like looking closely at one tree in the forest to see if it’s growing differently because of something happening around it. This helps us understand how individual genes respond to changes and can be important for things like developing new drugs.</p></li>
<li><p><strong>Multiple Genes Level:</strong> Zooming out a bit, scientists ask questions about groups of genes. They wonder if there are sets of genes that work together or are somehow connected. For example, they might investigate if a bunch of genes all change their behavior in response to a treatment compared to a control. It’s like stepping back and observing a whole section of trees in the forest to see if they’re all affected in the same way. This helps us see broader patterns in how genes are working together.</p></li>
<li><p><strong>Networks of Genes Level:</strong> Finally, when researchers zoom out even further, they’re looking at the big picture. They want to identify and understand the complex web of interactions between genes, which we call regulatory and signaling networks. It’s like studying the entire ecosystem of the forest, where you’re not just looking at individual trees but also how they all interact with each other, from the smallest plants to the largest animals. This level helps us grasp the underlying mechanisms that control gene behavior.</p></li>
</ol>
<p>So, in essence, scientists use different “microscopes” to study genes, starting from the tiniest details and gradually uncovering larger and more complex interactions. It’s a bit like exploring a forest, from individual trees to the entire ecosystem, to truly understand how everything works together in the world of genetics.</p>
</section>
<section id="gene-ontology-i.e.-go" class="level3" data-number="7.4.2">
<h3 data-number="7.4.2" class="anchored" data-anchor-id="gene-ontology-i.e.-go"><span class="header-section-number">7.4.2</span> Gene Ontology (i.e., GO)</h3>
<p>When scientists want to understand how genes are connected to the jobs they do in our bodies, they use something called “Gene Ontology,” or GO for short. Think of it as a giant encyclopedia for genes that helps us make sense of what they do.</p>
<p>Now, this GO encyclopedia isn’t just one big book; it’s divided into three separate parts, kind of like having different sections in a library:</p>
<ol type="1">
<li><p><strong>Biological Process:</strong> In this section, scientists describe what specific jobs or tasks genes are involved in when it comes to biological processes. It’s like reading about the roles of different workers in a big company.</p></li>
<li><p><strong>Molecular Function:</strong> Here, scientists explain the specific functions or abilities of genes at the molecular level. It’s like understanding the unique skills or talents of each gene, such as their ability to interact with other molecules.</p></li>
<li><p><strong>Cellular Component:</strong> This section focuses on where genes are located or where they do their work inside cells. It’s like finding out which departments or rooms in a company each worker belongs to.</p></li>
</ol>
<p>But that’s not all! There are also other ways to group genes based on their functions and properties. For example, we have pathways like KEGG, Metacyc, and Reactome, which are like different roads or routes genes can take in the body. We also have categories like protein families and even information about where genes are located on chromosomes, which is a bit like knowing the street addresses of different businesses in a city.</p>
</section>
<section id="how-does-gene-set-analysis-work" class="level3" data-number="7.4.3">
<h3 data-number="7.4.3" class="anchored" data-anchor-id="how-does-gene-set-analysis-work"><span class="header-section-number">7.4.3</span> How Does Gene Set Analysis Work?</h3>
<p>Imagine you’re trying to figure out if a group of friends are all good at playing a game, but you only have a few chances to watch them play. If you try to judge each friend’s skills one by one, it might be really hard to tell if they’re good or not because your observations could be a bit shaky.</p>
<p>Now, what if you look at how the group as a whole plays the game? It’s like putting all their skills together and looking at the big picture. This can make it easier to see if they’re doing well or not. That’s the basic idea behind analyzing gene sets.</p>
<p>When scientists have limited data, like not enough measurements for each individual gene, it can be tough to tell if a single gene is acting differently. But when they group genes together based on certain criteria (like genes that are related to a specific process), they can see if that whole group is changing in a noticeable way. It’s like looking at how the entire team of friends is playing the game rather than just focusing on one person. This approach helps scientists spot big changes in gene groups more easily.</p>
<p>This way of looking at things can help scientists find out if a bunch of genes are working together or changing together, kind of like friends who play the same game in a similar way. Plus, based on experimental evidence, it seems that these group effects are often consistent across different experiments, while individual gene effects can be a bit trickier to pin down and reproduce.</p>
</section>
</section>
<section id="enrichment-analysis" class="level2" data-number="7.5">
<h2 data-number="7.5" class="anchored" data-anchor-id="enrichment-analysis"><span class="header-section-number">7.5</span> Enrichment Analysis</h2>
<p>This is a way scientists figure out if certain groups of genes are doing something special in a given situation.</p>
<ol type="1">
<li><p><strong>Singular Enrichment Analysis:</strong> Imagine you have a list of genes that you think are acting differently under certain conditions. In this approach, you assume you already know which genes are different and you have a specific set of rules for what counts as a significant change. You then test each group of genes one by one to see if they’re involved in any special processes or functions. It’s like checking each item on a shopping list to see if it’s available at a store. This approach treats each group of genes independently.</p></li>
<li><p><strong>Gene Set Enrichment Analysis:</strong> Now, think of a scenario where you don’t have strict rules for what counts as a significant change. Instead, you compare the entire group of genes you’re interested in to lots of random groups to see if there’s something unique about them. It’s like comparing a special recipe to a bunch of random ones to see if it’s really unique and delicious. This approach doesn’t rely on specific thresholds; it’s more about comparing your group of genes to many random ones.</p></li>
<li><p><strong>Modular Enrichment Analysis:</strong> In this approach, you already have a list of genes that you think are behaving differently, but you’re not sure how they’re related. So, you group these genes together based on how they’re described or annotated, kind of like sorting them into different boxes based on their similarities. Then, you see if each box (or module) of genes is involved in any special processes or functions. It’s like organizing your closet by grouping similar clothes together and then checking each group to see if they have a unique style. This approach focuses on the relationships between genes based on their descriptions.</p></li>
</ol>
<section id="singular-enrichment-analysis" class="level3" data-number="7.5.1">
<h3 data-number="7.5.1" class="anchored" data-anchor-id="singular-enrichment-analysis"><span class="header-section-number">7.5.1</span> Singular Enrichment Analysis</h3>
<p>Imagine you have a group of 30 genes, and out of these, a whopping 24 are labeled as “ribosomal proteins.” It might seem like something important is happening with ribosomal proteins in your experiment, right?</p>
<p>But hold on, there’s a twist. What if you knew that this group of 30 genes was originally part of a much larger group of 100 genes, and out of those, 90 were also “ribosomal proteins”? Suddenly, it doesn’t seem as extraordinary, does it?</p>
<p>That’s where the idea of considering the background comes into play. You see, just knowing how many genes are related to a particular biological process isn’t enough. You need to compare it to what you’d expect by chance. Are you seeing more of those genes than you’d anticipate? Or is it just a typical result?</p>
<p>To figure this out, scientists use a statistical test. It’s like having a way to check if the number of “ribosomal proteins” you found in your group of 30 genes is statistically significant, meaning it’s not just due to chance. It helps ensure that when you say something is happening with a biological process, you’re not making a false alarm based on random fluctuations.</p>
<section id="fishers-exact-test" class="level4" data-number="7.5.1.1">
<h4 data-number="7.5.1.1" class="anchored" data-anchor-id="fishers-exact-test"><span class="header-section-number">7.5.1.1</span> Fisher’s Exact Test</h4>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../imgs/week7/lady_matrix.png" class="img-fluid figure-img" width="140"></p>
<p></p><figcaption class="figure-caption">Truth Matrix as Seen by Fisher</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>Let’s take a journey back in time to understand Fisher’s Exact Test and its interesting history. Imagine a time when people were fascinated by unusual claims, like Muriel Bristol’s assertion that she could tell whether milk or tea was poured first just by tasting it.</p>
<p><span class="math display">\[\begin{equation}
  p = \frac{(a+b)!(c+d)!(a+c)!(b+d)!}{n!a!b!c!d!}
\end{equation}\]</span></p>
<p>Now, enter Sir Ronald A. Fisher, a brilliant statistician. He wanted to put Muriel’s claim to the test in a scientific way. So, he came up with what’s known today as <strong>Fisher’s Exact Test</strong>. The idea behind this test was to determine if there was a significant connection between what Muriel called the “Truth” (the actual order of pouring milk or tea) and what “Lady says” (Muriel’s claims based on her taste).</p>
<section id="in-bioinformatics" class="level5" data-number="7.5.1.1.1">
<h5 data-number="7.5.1.1.1" class="anchored" data-anchor-id="in-bioinformatics"><span class="header-section-number">7.5.1.1.1</span> In Bioinformatics</h5>
<p>Now, let’s dive into how Fisher’s Exact Test is used in the world of gene sets, specifically in the context of gene expression analysis. Imagine you have two categories that you want to compare:</p>
<ol type="1">
<li><p><strong>“Gene belongs to the (GO/KEGG) category”:</strong> This category is all about genes that have specific labels or annotations, like being part of a certain biological process or pathway.</p></li>
<li><p><strong>“Gene is in the list of differentially expressed genes”:</strong> This category includes genes that you’ve identified as behaving differently under certain experimental conditions.</p></li>
</ol>
<p>Now, what Fisher’s Exact Test does here is pretty clever. It examines whether these two categories significantly overlap or not. In simpler terms, it helps you figure out if the genes with specific labels or annotations (the first category) tend to show up more often in the list of genes that behave differently (the second category) than you’d expect by random chance.</p>
<p>It’s like being a detective once again. You’re trying to find out if there’s a real connection between certain types of genes (those with specific labels) and genes that change their behavior under certain conditions. Fisher’s Exact Test helps you determine if this connection is more than just a coincidence, providing a statistical way to make sense of the data and draw meaningful conclusions in gene expression analysis.</p>
</section>
<section id="pros-and-cons" class="level5" data-number="7.5.1.1.2">
<h5 data-number="7.5.1.1.2" class="anchored" data-anchor-id="pros-and-cons"><span class="header-section-number">7.5.1.1.2</span> Pros and Cons</h5>
<p>Let’s explore the benefits and downsides of Fisher’s Exact Test, which is like a versatile tool in the world of statistics, but it comes with its own strengths and limitations.</p>
<ol type="1">
<li><p><strong>Benefits of Fisher’s Exact Test:</strong></p>
<ol type="1">
<li><p><strong>Simple and Fast:</strong> One of its standout features is that it’s quite straightforward to use. You don’t need to be a statistics expert to apply it. Plus, it’s speedy, providing results quickly. It’s like having a simple and efficient tool in your toolkit.</p></li>
<li><p><strong>Good Performance:</strong> Fisher’s test already does a pretty good job at what it’s designed for. It’s like having a reliable and well-tuned instrument that usually gives you meaningful results.</p></li>
<li><p><strong>Versatility:</strong> You can use it in various situations and contexts. It’s not limited to a specific type of data or research area. Think of it as a Swiss Army knife in statistics that can be applied in different scenarios.</p></li>
</ol></li>
<li><p><strong>Downsides of Fisher’s Test:</strong></p>
<ol type="1">
<li><p><strong>Setting a Threshold:</strong> To use Fisher’s test effectively, you need to decide on a threshold for what counts as “differential expression.” This can sometimes be a bit subjective and may vary depending on the specific experiment.</p></li>
<li><p><strong>Multiple P-Values:</strong> When you apply Fisher’s test to multiple categories (like different gene sets), you end up with a bunch of p-values, one for each category. This can be overwhelming if you have hundreds of enriched categories with p-values less than 0.05. It’s like having a stack of puzzle pieces to put together.</p></li>
<li><p><strong>Correction for Multiple Testing:</strong> Because you’re dealing with lots of p-values, you often need to correct for multiple testing. This step ensures that you don’t mistakenly think something is significant just by chance. However, when categories overlap, it can be challenging to account for this in the correction process.</p></li>
<li><p><strong>Different Gene Set Sizes:</strong> Not all gene sets are the same size. Some may have many genes, while others have only a few. Fisher’s test might treat them all equally, which can be a limitation when you’re comparing different sets. It’s like trying to compare the impact of big and small puzzle pieces without adjusting for their size.</p></li>
</ol></li>
</ol>
<p>So, Fisher’s Exact Test is like a trusty tool that gets the job done quickly and effectively in many situations. But, like any tool, it has its quirks, such as the need for threshold decisions and dealing with multiple p-values, making it important to use it thoughtfully and consider its limitations.</p>
</section>
</section>
</section>
</section>
<section id="gene-set-enrichment" class="level2" data-number="7.6">
<h2 data-number="7.6" class="anchored" data-anchor-id="gene-set-enrichment"><span class="header-section-number">7.6</span> Gene Set Enrichment</h2>
<p>Imagine you’re looking at a big collection of genes, and you suspect that there might be small but coordinated changes happening in their expression levels. But here’s the thing: you don’t want to decide on a specific rule or threshold for what counts as a “big” change because it could vary from one experiment to another.</p>
<p>That’s where gene set enrichment tests come into play. The basic idea behind these tests is to do two things:</p>
<ol type="1">
<li><p><strong>Compute a Gene Set-Wise Test Statistic:</strong> For each category or group of genes (like those related to a specific biological process), you calculate a special number called a “test statistic.” This number summarizes how the genes in that category are behaving, kind of like a score that tells you if something interesting might be going on.</p></li>
<li><p><strong>Compare to Random Samples:</strong> Next, you compare the test statistic you calculated for the actual group of genes to what you’d expect if you randomly picked genes. It’s like checking if the score for your group of genes is way different from what you’d get by just picking genes at random.</p></li>
</ol>
<p>If the test statistic for your group of genes stands out from what you’d expect by chance, it suggests there’s something significant happening in that category. It’s like finding a treasure chest among a bunch of empty boxes.</p>
<section id="gene-set-enrichment-analysis" class="level3" data-number="7.6.1">
<h3 data-number="7.6.1" class="anchored" data-anchor-id="gene-set-enrichment-analysis"><span class="header-section-number">7.6.1</span> Gene Set Enrichment Analysis</h3>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../imgs/week7/gsea_workflow.png" class="img-fluid figure-img" width="186"></p>
<p></p><figcaption class="figure-caption">General Workflow for GSEA</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>Let’s simplify the process of Gene Set Enrichment Analysis (GSEA) so that it’s easier to understand:</p>
<ol type="1">
<li><p><strong>Sorting Genes:</strong> First, we line up all the genes based on how differently they’re behaving in an experiment. Some genes might be more active, while others are less active.</p></li>
<li><p><strong>Calculating Enrichment Score (ES):</strong> Now, we want to figure out if certain groups of genes, let’s call them “gene sets,” are doing something special. We go through the list of genes and do two things for each gene:</p>
<ul>
<li>If the gene belongs to a particular gene set we’re interested in (let’s call it “G”), we add 1 to our score.</li>
<li>If it doesn’t belong to that gene set, we subtract 1 from our score. After we’ve done this for all the genes, we find the highest score, which we call the “maximum enrichment score” (MES).</li>
</ul></li>
<li><p><strong>Testing It:</strong> Imagine we want to make sure our results aren’t just by chance. So, we create a “null hypothesis” by doing a pretend experiment 1,000 times:</p>
<ul>
<li>We mix up the labels on our measurements (like shuffling cards).</li>
<li>Then, we repeat steps 1 and 2, calculating the MES for this shuffled data.</li>
</ul></li>
<li><p><strong>Checking Significance:</strong> Finally, we ask a question: “How many times did our pretend MES (from the shuffled data) turn out to be as extreme as or even more extreme than the real MES we found in our actual data?”</p>
<ul>
<li>If the real MES is way more extreme than what we got in our pretend experiments, it suggests that the gene set we’re looking at is probably doing something important.</li>
</ul></li>
</ol>
<p>So, in simple terms, GSEA helps us find out if certain groups of genes are acting differently in our experiment compared to what we’d expect by random chance. If they are, it could mean they’re involved in something significant.</p>
</section>
<section id="gene-set-analyses" class="level3" data-number="7.6.2">
<h3 data-number="7.6.2" class="anchored" data-anchor-id="gene-set-analyses"><span class="header-section-number">7.6.2</span> Gene Set Analyses</h3>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../imgs/week7/gene_set_analysis.png" class="img-fluid figure-img" width="154"></p>
<p></p><figcaption class="figure-caption">Arrays and Genes in GSA</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>Let’s break down Gene Set Analysis (GSA) in a way that’s easy to understand:</p>
<ol type="1">
<li><p><strong>Gene Test Statistics:</strong> First, we look at each gene individually and calculate a special number called a “test statistic.” This number helps us understand how different each gene is behaving in our experiment. Think of it as a score for each gene.</p></li>
<li><p><strong>Max-Mean for Sets:</strong> Next, we’re interested in groups of genes, which we call “sets.” Instead of looking at each gene on its own, we want to know if whole sets of genes are doing something special. To figure this out, we find the maximum average score <span class="math inline">\(\max\{\bar{s}^{(+)}, \bar{s}^{(-)}\}\)</span> for each set. It’s like looking for the highest average score among a group of friends playing a game.</p></li>
<li><p><strong>Standardizing the Scores:</strong> We want to make sure our results are reliable, so we do a bit of math to make everything fair and comparable: <span class="math inline">\(\displaystyle S'_{\text{max}} = \frac{(S_\text{max} - \mu_S)}{\sigma_S}\)</span>. This step involves standardizing the scores to put them on the same scale, so we can easily compare different sets of genes.</p></li>
<li><p><strong>Shuffling the Data:</strong> To be extra sure that our findings are meaningful and not just random, we do some pretend experiments. We mix up the data by shuffling the columns around, kind of like playing cards. Then, we calculate the max-mean scores for these shuffled data sets many times (let’s say B times).</p></li>
</ol>
<p>As an extra tidbit of information, the p-values for the <span class="math inline">\(S'_\text{max}\)</span> shown above can be calculated using the following formula:</p>
<p><span class="math display">\[\begin{equation}
  p_s = \frac{\#(S'_\text{max} &lt; S^{'B})}{B}
\end{equation}\]</span></p>
<p>So, in a nutshell, Gene Set Analysis helps us figure out if groups of genes are working together in a special way in our experiment. We start by looking at each gene’s behavior, then we check if sets of genes have higher average scores. We make sure our results are reliable by standardizing them and doing some pretend experiments with shuffled data. It’s like investigating whether certain teams of friends are winning a game together, but we want to be sure it’s not just luck!</p>
<section id="problems-with-gene-set-analysis" class="level4" data-number="7.6.2.1">
<h4 data-number="7.6.2.1" class="anchored" data-anchor-id="problems-with-gene-set-analysis"><span class="header-section-number">7.6.2.1</span> Problems with Gene Set Analysis</h4>
<p>Let’s simplify the challenges in gene set enrichment tests:</p>
<ol type="1">
<li><p><strong>Summarizing Gene Data:</strong> In these tests, we need to crunch down all the information about each gene into one simple number. This can be tricky, especially in complex experiments with lots of data points. It’s like trying to describe a whole book with just one word.</p></li>
<li><p><strong>Missing Important Clues:</strong> These tests might not catch every important detail about how genes work together. Imagine some pathways in our body are like a security system with gatekeeper genes that control what happens. Just looking at the overall expression of genes might not tell us if these gatekeepers are active or not.</p></li>
</ol>
<p>So, while gene set enrichment tests are helpful, they have their limitations. They sometimes struggle to give us a clear picture of what’s happening in complex experiments, and they might miss some important clues about gene activity. It’s a bit like trying to understand a big puzzle, and we might need some extra pieces to see the whole picture. That’s all for today!</p>


</section>
</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  let localAlternateSentinel = 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } 
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../chapters/week6.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Transcriptomes: Experiment Design and RNA Sequence Data Processing</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../chapters/week8.html" class="pagination-link">
        <span class="nav-page-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Transcriptomes: Network Analysis</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->



</body></html>