# Comparative Genomics

When examining genomes, the analysis occurs at different levels to understand both the variations within a species for adaptation and the evolutionary differences between species. 

Within a species, researchers focus on smaller-scale differences known as Single Nucleotide Polymorphisms (SNPs), which are tiny genetic variations at the single nucleotide level. These SNPs provide insights into population genetics, allowing scientists to study genetic diversity and adaptations within a population. On a larger scale, structural variants such as insertions, deletions, and rearrangements are explored. These structural variants are a part of pan-genome studies, which aim to uncover genetic diversity beyond the core genome of a species.

When comparing genomes between different species, the emphasis is on evolutionary analysis. This involves investigating the evolution of genes or gene families across species to comprehend their divergence or conservation. Additionally, researchers construct phylogenies or evolutionary trees to illustrate the relationships between species based on genetic similarities and differences.

Various methods are employed in comparing genomes. **Synteny analysis** focuses on examining the order and arrangement of genes along chromosomes, providing insights into how genetic material is conserved or rearranged across species. The study of gene content, involving gene families and their expansions or contractions, sheds light on the evolutionary changes in genetic functions. Moreover, genomic alignment is utilized to align entire genomes, revealing regions of high conservation (ultraconserved regions) or areas susceptible to recombination (recombination hotspots).

```{r echo = F, fig.cap = 'Illustration of Molecular Phylogeny', fig.align = 'center'}
knitr::include_graphics('../imgs/week12/molecular_phylogeny.png')
```

**Molecular phylogeny** is a method used to infer taxonomic relationships among species by analyzing genomic data. It involves several steps, including multiple sequence alignment and the estimation of gene trees.

Through multiple sequence alignment, researchers compare genetic sequences from different species to identify similarities and differences. This process helps in estimating gene trees, which illustrate the evolutionary relationships between different genes or genomic regions across species.

**Species phylogeny**, inferred from these gene trees, reflects mutations that have reached fixation in distinct species over evolutionary time. Fixed mutations are alterations in the genetic code that have become prevalent and stable within specific species populations, distinguishing them from other related species.

In the context of identifying conserved regions in proteins, molecular phylogeny contributes to recognizing areas within proteins that remain highly conserved across species. These conserved regions are important because they often perform essential functions and have persisted throughout evolutionary history due to their significance.

## Substitution Rates and Molecular Clocks

```{r echo = F, fig.cap = 'Mutations', fig.align = 'center'}
knitr::include_graphics('../imgs/week12/mutations.png')
```

When an organism has a mutation, it's usually passed down from their generation to the next (assuming that they have offspring).

### Comparing Sequences

```{r echo = F, fig.cap = 'Sequences', fig.align = 'center'}
knitr::include_graphics('../imgs/week12/sequences.png')
```

Dynamic programming algorithms, such as the Needleman-Wunsch algorithm, offer a global solution for pairwise sequence alignment, crucial in bioinformatics. Dotplots are graphical representations aiding in sequence alignments.

In constructing a dotplot:

1. **Grid Representation:** Imagine a grid where sequences are placed on the X and Y axes. The grid starts in the upper left corner and concludes at the lower right corner.

2. **Movement Options:** The algorithm allows three possible moves: right, down, or diagonally down-right. These movements within the grid help build the alignment.

3. **Scoring Matches and Gaps:** Passing through a cell containing matching letters elevates the alignment score, contributing to the similarity of the sequences. Conversely, moving through an empty cell introduces a gap, incurring a penalty that impacts the alignment score.

Dotplots visually demonstrate matches (dots) in the grid where sequences align. A **dot** is plotted when a cell in the grid signifies a match, indicating similar segments between the sequences. Dotplots assist in identifying regions of similarity, divergence, gaps, or duplications between sequences, aiding biologists in understanding genetic relationships and structural similarities between DNA or protein sequences.

However, one key thing to note about this approach is that it is an NP-complete problem.
